# Data statistics

## Sequencing reads statistics

```{r load_data_stats}
load("data/data.Rdata")
```

### Loading data

```{r load preprocessing data}
preprocessing <- read_tsv("data/preprocessing.tsv")
```
Total number of bases sequenced (Gb)
```{r reads_stats}
preprocessing %>% 
    summarise(Total=sum(reads_metagenomic * 150 / 1000000000) %>% round(2), 
              mean=mean(reads_metagenomic * 150 / 1000000000) %>% round(2),
              sd=sd(reads_metagenomic * 150 / 1000000000) %>% round(2)) %>%
    unite("Average",mean, sd, sep = " ± ", remove = TRUE) %>%
    tt()
```


How many samples do we have of each group (behaviour)?
```{r checking samples}
sample_metadata %>%
  group_by(fox_behaviour) %>%
  summarize(n_samples = n()) %>%
  tt()
```
How many MAGs?
```{r number of genomes}
genome_metadata %>%
  summarize(n_genomes = n())%>%
  tt()
```


```{r mean completeness and contamination}
genome_metadata %>% 
  summarise(
    mean_c = mean(completeness, na.rm = TRUE) %>% round(2),
    sd_c = sd(completeness, na.rm = TRUE) %>% round(2),
    mean_con = mean(contamination, na.rm = TRUE) %>% round(2),
    sd_con = sd(contamination, na.rm = TRUE) %>% round(2)
  ) %>%
  unite("Completeness", mean_c, sd_c, sep = " ± ") %>%
  unite("Contamination", mean_con, sd_con, sep = " ± ") %>%
  tt()
```


### DNA fractions

```{r sequence_fractions_calculation}
sequence_fractions <- read_counts %>%
  pivot_longer(-genome, names_to = "sample", values_to = "value") %>%
  group_by(sample) %>%
  summarise(mags = sum(value)) %>%
	left_join(preprocessing, by = join_by(sample == sample)) %>%
	dplyr::select(sample,mags,bases_metagenomic,bases_host ,bases_discarded, bases_raw) %>%
	mutate(mags_bases = mags*146) %>% #total number of bases mapped to mags (multiplying by effective read length)
  mutate(bases_discarded_proportion = bases_discarded/bases_raw) %>%
	mutate(lowqual_bases = ((bases_metagenomic+bases_host)/(1-bases_discarded_proportion))-(bases_metagenomic+bases_host)) %>%
	mutate(unmapped_bases = bases_metagenomic - mags_bases) %>%
	mutate(unmapped_bases = ifelse(unmapped_bases < 0, 0, unmapped_bases)) 
```

```{r sequence_fractions_gb}
sequence_fractions %>%
	dplyr::select(sample, lowqual_bases, bases_host, unmapped_bases, mags_bases)  %>%
  mutate_at(vars(-sample), ~./1000000000) %>%
  rename("Sample"=1, "Low quality"=2, "Mapped to host"=3, "Unmapped"=4, "Mapped to MAGs"=5) %>%
  tt()
```


```{r dna_fractions_mean}
#mean
sequence_fractions %>%
	dplyr::select(sample, lowqual_bases, bases_host, unmapped_bases, mags_bases)  %>%
  mutate_at(vars(-sample), ~./1000000000) %>%
  rename("Sample"=1, "Low quality"=2, "Mapped to host"=3, "Unmapped"=4, "Mapped to MAGs"=5) %>%
  summarise(across(where(is.numeric), mean, na.rm = TRUE)) %>%
  tt()
```

```{r sequence_fractions_sd}
#SD
sequence_fractions %>%
	dplyr::select(sample, lowqual_bases, bases_host, unmapped_bases, mags_bases)  %>%
  mutate_at(vars(-sample), ~./1000000000) %>%
  rename("Sample"=1, "Low quality"=2, "Mapped to host"=3, "Unmapped"=4, "Mapped to MAGs"=5) %>%
  summarise(across(where(is.numeric), sd, na.rm = TRUE)) %>% 
  tt()
```



```{r dna_fractions_plot, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
sequence_fractions_plot <- sequence_fractions %>%
  dplyr::select(sample, lowqual_bases, bases_host, unmapped_bases, mags_bases)  %>%
	pivot_longer(!sample, names_to = "fraction", values_to = "value") %>%
	mutate(value = value / 1000000000) %>%
	mutate(fraction = factor(fraction, levels = c("lowqual_bases","bases_host","unmapped_bases","mags_bases"))) 
  
ggplot(sequence_fractions_plot, aes(x = sample, y = value, fill=fraction)) +
    geom_bar(position="stack", stat = "identity") +
    scale_fill_manual(name="Sequence type",
                  breaks=c("lowqual_bases","bases_host","unmapped_bases","mags_bases"),
                  labels=c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                  values=c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c"))+
    labs(x = "Samples", y = "Amount of data (GB)") +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size=6),legend.position = "bottom")
```


```{r dna_fractions_plot_sample_type, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
sequence_fractions_type <- sequence_fractions_plot %>%
  left_join(sample_metadata) 

ggplot(sequence_fractions_type, aes(x = sample, y = value, fill = fraction)) +
    geom_bar(position = "stack", stat = "identity") +
    scale_fill_manual(name = "Sequence type",
                      breaks = c("lowqual_bases","bases_host","unmapped_bases","mags_bases"),
                      labels = c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                      values = c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c")) +
    labs(x = "Samples", y = "Amount of data (GB)") +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 6),
          legend.position = "bottom") +
    facet_wrap(~ sample_type, scales = "free_x", nrow = 1)
    
```



```{r dna_fractions_plot_gut_location, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(sequence_fractions_type, aes(x = sample, y = value, fill = fraction)) +
    geom_bar(position = "stack", stat = "identity") +
    scale_fill_manual(name = "Sequence type",
                      breaks = c("lowqual_bases","bases_host","unmapped_bases","mags_bases"),
                      labels = c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                      values = c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c")) +
    labs(x = "Samples", y = "Amount of data (GB)") +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 6),
          legend.position = "bottom") +
    facet_wrap(~ gut_location, scales = "free_x", nrow = 1)
    
```


```{r dna_fractions_plot_sample_types, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(sequence_fractions_type, aes(x = sample, y = value, fill = fraction)) +
    geom_bar(position = "stack", stat = "identity") +
    scale_fill_manual(name = "Sequence type",
                      breaks = c("lowqual_bases","bases_host","unmapped_bases","mags_bases"),
                      labels = c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                      values = c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c")) +
       labs(x = "Samples", y = "Amount of data (GB)") +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 6),
          legend.position = "bottom") +
    facet_nested(~ sample_type + gut_location,
                 scales = "free_x", space = "free_x")

```



```{r dna_fractions_plot_fox_behaviour, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(sequence_fractions_type, aes(x = sample, y = value, fill = fraction)) +
    geom_bar(position = "stack", stat = "identity") +
    scale_fill_manual(name = "Sequence type",
                      breaks = c("lowqual_bases","bases_host","unmapped_bases","mags_bases"),
                      labels = c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                      values = c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c")) +
    labs(x = "Samples", y = "Amount of data (GB)") +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 6),
          legend.position = "bottom") +
    facet_wrap(~ fox_behaviour, scales = "free_x", nrow = 1)
    
```

## Filtering
To remove samples with high host data or low metagenomic data
```{r host fraction plot}
host_data_plot <- sequence_fractions %>%
  left_join(sample_metadata) %>%
  mutate(host_fraction = bases_host / bases_raw,
         sample = reorder(sample, -host_fraction))  #reorder by decreasing host_fraction

ggplot(host_data_plot,aes(x = sample, y = host_fraction, fill = gut_location)) +
  geom_col() +
  scale_fill_manual(
  values = c(
    F_colon = "#d6604d",
    M_colon = "#542788",
    D_ileum = "#fdb863",
    K_ileum = "#bb99d8"
  )
)+
  geom_hline(yintercept = 0.5, linetype = "dashed", color = "black")+
  labs(y = "Host fraction", x = "Sample",
       title = "Fraction of host bases per sample")+
   theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1)  
  )
```


```{r samples with high host data}
threshold <- 0.5

samples_to_remove <- host_data_plot %>%
  filter(host_fraction > threshold) %>%
  pull(sample)

samples_to_remove
```


```{r MAG gb plot}
host_data_plot <- sequence_fractions %>%
  left_join(sample_metadata) %>%
  mutate(host_fraction = bases_host / bases_raw,
                bases_metagenomic_GB = bases_metagenomic / 1e9, sample = reorder(sample, -bases_metagenomic))  #reorder by decreasing bases

ggplot(host_data_plot,aes(x = sample, y = bases_metagenomic_GB, fill = gut_location)) +
  geom_col() +
  scale_fill_manual(
  values = c(
    F_colon = "#d6604d",
    M_colon = "#542788",
    D_ileum = "#fdb863",
    K_ileum = "#bb99d8"
  )
)+
  geom_hline(yintercept = 1, linetype = "dashed", color = "black")+
  labs(y = "Metagenomic bases (Gb)", x = "Sample",
       title = "GB of metagenomic bases per sample")+
   theme_bw() +
  scale_y_continuous(
    breaks = seq(0, max(host_data_plot$bases_metagenomic_GB), by = 1)
  ) +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1)  
  )
```




```{r host data vs metagenomic reads plot}
ggplot(host_data_plot, aes(x= host_fraction, y= bases_metagenomic_GB, color = gut_location))+
   scale_color_manual(
  values = c(
    F_colon = "#d6604d",
    M_colon = "#542788",
    D_ileum = "#fdb863",
    K_ileum = "#bb99d8"
  ))+
  scale_y_continuous(
    breaks = seq(0, max(host_data_plot$bases_metagenomic_GB), by = 1)
  ) +
  geom_hline(yintercept = 1, linetype = "dashed", color = "black")+
  geom_vline(xintercept = 0.75, linetype = "dashed", color = "black")+
  labs(y = "Metagenomic Bases", x = "Host fraction")+
  geom_point()+
  theme_bw()
```

```{r samples with low metagenomic data}
samples_to_remove_total <- host_data_plot %>%
  filter(bases_metagenomic_GB < 1) %>%  #remove samples with less than 1Gb of metagenomic reads
   filter(host_fraction > 0.75) %>% #remove samples with more than 0.75 host fraction
  pull(sample)

samples_to_remove_total

#Add samples to remove to the Rdata
save(samples_to_remove_total,  file = "data.RData")
```


```{r filtering and plotting again}
host_data_filtered <- host_data_plot %>%
  filter(!sample %in% samples_to_remove_total) %>%
  mutate(host_fraction = bases_host / bases_raw,
         sample = reorder(sample, -host_fraction)) 

ggplot(host_data_filtered,aes(x = sample, y = host_fraction, fill = gut_location)) +
  geom_col() +
  scale_fill_manual(
  values = c(
    F_colon = "#d6604d",
    M_colon = "#542788",
    D_ileum = "#fdb863",
    K_ileum = "#bb99d8"
  )
)+
  geom_hline(yintercept = 0.5, linetype = "dashed", color = "black")+
  labs(y = "Host fraction", x = "Sample",
       title = "Fraction of host bases per sample")+
   theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1)  
  )
```



