# Community composition

```{r load_data_community, echo=FALSE}
load("data/data.Rdata")
```
```{r behaviour colors}
behaviour_colors <- c(
  tame = "#1f77b4",   
  aggr = "#d62728",   
  unsel = "#2ca02c"   
)
```

## Taxonomy overview 

### Stacked barplot

```{r preparing data for bar plots}
tax_barplot <- genome_counts_filt %>%
  mutate_at(vars(-genome), ~./sum(.)) %>% #apply TSS normalization
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0)  #filter 0 counts
```

```{r fitlered data according to host fraction and metagenomic bases}
tax_barplot_filt <- genome_counts_filt %>%
 dplyr::select(-all_of(samples_to_remove))%>%
  mutate_at(vars(-genome), ~./sum(.)) %>% #apply TSS normalization
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0)  #filter 0 counts
```


Evaluating the community composition at each of the gut locations.
```{r taxonomy_barplot_gut_location, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(tax_barplot, aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_grid(. ~ gut_location,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 6),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")

```

```{r taxonomy_barplot_gut_location_filtered, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(tax_barplot_filt, aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_grid(. ~ gut_location,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 6),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")

```

Evaluating community composition between the fox behaviours within the different gut locations:
```{r taxonomy_barplot, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(tax_barplot, aes(x=sample,y=count, fill=phylum, group=phylum)) + 
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~ gut_location + fox_behaviour,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```

```{r taxonomy_barplot_2, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(tax_barplot, aes(x=sample,y=count, fill=phylum, group=phylum)) + 
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~  fox_behaviour + gut_location,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```

```{r taxonomy_barplot_3, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(tax_barplot, aes(x=sample,y=count, fill=phylum, group=phylum)) + 
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~  fox_behaviour + sample_type,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```

```{r taxonomy_barplot_4, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(tax_barplot, aes(x=sample,y=count, fill=phylum, group=phylum)) + 
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~   sample_type+ fox_behaviour,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```
### Only with F_colon
```{r keeping only sample from F_colon}
tax_barplot_fcolon <- tax_barplot %>%
  filter(gut_location == "F_colon")
```


```{r taxonomy_barplot_fcolon, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(tax_barplot_fcolon, aes(x=sample,y=count, fill=phylum, group=phylum)) + 
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~  fox_behaviour,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```



### Phylum relative abundances

```{r taxonomy_phylum_summary, warning=FALSE, comments="", message=FALSE}
phylum_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalization
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,phylum,fox_behaviour) %>%
  summarise(relabun=sum(count))

phylum_summary %>%
    group_by(phylum) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              tame_mean=mean(relabun[fox_behaviour =="tame"]*100, na.rm=T),
              tame_sd=sd(relabun[fox_behaviour =="tame"]*100, na.rm=T),
              aggressive_mean=mean(relabun[fox_behaviour =="aggr"]*100, na.rm=T),
              aggressive_sd=sd(relabun[fox_behaviour=="aggr"]*100, na.rm=T),
              unsel_mean=mean(relabun[fox_behaviour =="unsel"]*100, na.rm=T),
              unsel_sd=sd(relabun[fox_behaviour =="unsel"]*100, na.rm=T)) %>%
    mutate(total=str_c(round(total_mean,2),"±",round(total_sd,2)),
           tame=str_c(round(tame_mean,2),"±",round(tame_sd,2)),
           aggresive=str_c(round(aggressive_mean,2),"±",round(aggressive_sd,2)),
           unselected =str_c(round(unsel_mean,2),"±",round(unsel_sd,2))) %>% 
    arrange(-total_mean) %>% 
   dplyr::select(phylum,total,tame,aggresive, unselected) %>% 
    tt()
```


Distribution of relative abundance across samples:
```{r taxonomy_phylum_plot, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_arrange <- phylum_summary %>%
    group_by(phylum) %>%
    summarise(mean=mean(relabun)) %>%
    arrange(-mean) %>%
    dplyr::select(phylum) %>%
    pull()

phylum_summary %>%
  filter(phylum %in% phylum_arrange) %>%
  mutate(phylum=factor(phylum,levels=rev(phylum_arrange))) %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
    filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=phylum, group=phylum, color=phylum, fill=phylum)) +
  scale_color_manual(values=phylum_colors[rev(phylum_arrange)]) +
        scale_fill_manual(values=phylum_colors[-8]) +
        geom_boxplot(alpha=0.2)+
        geom_jitter(alpha=0.5) + 
        theme_minimal() + 
        theme(legend.position="none") +
        labs(y="Phylum",x="Relative abundance")
```

```{r taxonomy_boxplot_phylum, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
## Calculate the phylum order based on mean relative abundance
phylum_arrange <- phylum_summary %>%
    group_by(phylum) %>%
    summarise(mean=mean(relabun)) %>%
    arrange(-mean) %>%
    dplyr::select(phylum) %>%
    pull()

phylum_summary %>%
  filter(phylum %in% phylum_arrange) %>%
  mutate(phylum=factor(phylum,levels=rev(phylum_arrange))) %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
    filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=phylum, group=phylum, color=phylum, fill=phylum)) +
  scale_color_manual(values=phylum_colors[rev(phylum_arrange)]) +
        scale_fill_manual(values=phylum_colors[-8]) +
        geom_boxplot(alpha=0.2)+
        geom_jitter(alpha=0.5) + 
        facet_nested(. ~ gut_location)+ 
        theme_minimal() + 
        theme(legend.position="none",
              strip.text.x = element_text(size = 14, color="black",face="bold"),
              axis.text.x = element_text(vjust = 0.5, size = 10),
              axis.text.y = element_text(size = 12),
              axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
              axis.title = element_text(size = 14, face = "bold"),
              axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
              axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
              ) +
        labs(y="Phylum",x="Relative abundance")
```




```{r taxonomy_boxplot_phylum_behaviour, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_summary %>%
  filter(phylum %in% phylum_arrange) %>%
  mutate(phylum=factor(phylum,levels=rev(phylum_arrange))) %>%
    filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=phylum, group=phylum, color=phylum, fill=phylum)) +
  scale_color_manual(values=phylum_colors[rev(phylum_arrange)]) +
        scale_fill_manual(values=phylum_colors[-8]) +
        geom_boxplot(alpha=0.2)+
        geom_jitter(alpha=0.5) + 
        facet_grid(.~fox_behaviour)+
        theme_minimal() + 
        theme(legend.position="none",
              strip.text.x = element_text(size = 14, color="black",face="bold"),
              axis.text.x = element_text(vjust = 0.5, size = 10),
              axis.text.y = element_text(size = 12),
              axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
              axis.title = element_text(size = 14, face = "bold"),
              axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
              axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
              ) +
        labs(y="Phylum",x="Relative abundance")


```


### Family

```{r family_all, comment="", echo=FALSE}
family_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalization
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,family,fox_behaviour) %>%
  summarise(relabun=sum(count))

family_arrange <- family_summary %>%
    group_by(family) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    dplyr::select(family) %>%
    pull()

family_summary %>%
    group_by(family) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              tame_mean=mean(relabun[fox_behaviour=="tame"]*100, na.rm=T),
              tame_sd=sd(relabun[fox_behaviour=="tame"]*100, na.rm=T),
              aggr_mean=mean(relabun[fox_behaviour=="aggr"]*100, na.rm=T),
              aggr_sd=sd(relabun[fox_behaviour=="aggr"]*100, na.rm=T),
              unsel_mean=mean(relabun[fox_behaviour =="unsel"]*100, na.rm=T),
              unsel_sd=sd(relabun[fox_behaviour =="unsel"]*100, na.rm=T)) %>%
    mutate(total=str_c(round(total_mean,2),"±",round(total_sd,2)),
           tame=str_c(round(tame_mean,2),"±",round(tame_sd,2)),
           aggr=str_c(round(aggr_mean,2),"±",round(aggr_sd,2)),
           unsel =str_c(round(unsel_mean,2),"±",round(unsel_sd,2))) %>% 
    arrange(-total_mean) %>% 
   dplyr::select(family,total,tame,aggr, unsel) %>% 
    tt()

```




```{r taxonomy_jitterplot_family_location, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
family_summary %>%
  left_join(genome_metadata %>% dplyr::select(family,phylum) %>% unique(),by=join_by(family==family)) %>%
  left_join(sample_metadata,by=join_by(sample==sample)) %>%
  filter(family %in% family_arrange[1:20]) %>%
  mutate(family=factor(family,levels=rev(family_arrange[1:20]))) %>%
  filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=family, group=family, color=phylum, fill=phylum)) +
      scale_color_manual(values=phylum_colors[-8]) +
      scale_fill_manual(values=phylum_colors[-8]) +
      geom_boxplot(alpha=0.2)+
      geom_jitter(alpha=0.5) + 
      facet_grid(.~gut_location)+
      theme_minimal() +
      theme(legend.position="none",
                  strip.text.x = element_text(size = 14, color="black",face="bold"),
                  axis.text.x = element_text(vjust = 0.5, size = 6),
                  axis.text.y = element_text(size = 12),
                  axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
                  axis.title = element_text(size = 14, face = "bold"),
                  axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
                  axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)))+
      labs(y="Family", x="Relative abundance", color="Phylum")
```

```{r taxonomy_jitterplot_family_fox_behaviour, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
family_summary %>%
    left_join(genome_metadata %>% dplyr::select(family,phylum) %>% unique(),by=join_by(family==family)) %>%
    left_join(sample_metadata,by=join_by(sample==sample)) %>%
    filter(family %in% family_arrange[1:20]) %>%
    mutate(family=factor(family,levels=rev(family_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=family, group=family, color=phylum, fill=phylum)) +
        scale_color_manual(values=phylum_colors[-8]) +
        scale_fill_manual(values=phylum_colors[-8]) +
        geom_boxplot(alpha=0.2)+
        geom_jitter(alpha=0.5) + 
        facet_grid(.~fox_behaviour.x)+
        theme_minimal() +
        theme(legend.position="none",
                    strip.text.x = element_text(size = 14, color="black",face="bold"),
                    axis.text.x = element_text(vjust = 0.5, size = 6),
                    axis.text.y = element_text(size = 12),
                    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
                    axis.title = element_text(size = 14, face = "bold"),
                    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
                    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)))+
        labs(y="Family", x="Relative abundance", color="Phylum")
```

### Genus

```{r genus_all, comment="", echo=FALSE}
genus_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalization
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,genus,fox_behaviour) %>%
  summarise(relabun=sum(count))

genus_arrange <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=sum(relabun)) %>%
    filter(genus != "g__")%>%
    arrange(-mean) %>%
    dplyr::select(genus) %>%
    mutate(genus= sub("^g__", "", genus)) %>%
    pull()

genus_summary %>%
    group_by(genus) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              tame_mean=mean(relabun[fox_behaviour=="tame"]*100, na.rm=T),
              tame_sd=sd(relabun[fox_behaviour=="tame"]*100, na.rm=T),
              aggr_mean=mean(relabun[fox_behaviour=="aggr"]*100, na.rm=T),
              aggr_sd=sd(relabun[fox_behaviour=="aggr"]*100, na.rm=T),
              unsel_mean=mean(relabun[fox_behaviour =="unsel"]*100, na.rm=T),
              unsel_sd=sd(relabun[fox_behaviour =="unsel"]*100, na.rm=T)) %>%
    mutate(total=str_c(round(total_mean,2),"±",round(total_sd,2)),
           tame=str_c(round(tame_mean,2),"±",round(tame_sd,2)),
           aggr=str_c(round(aggr_mean,2),"±",round(aggr_sd,2)),
           unsel = str_c(round(unsel_mean,2),"±",round(unsel_sd,2))) %>% 
    arrange(-total_mean) %>% 
   dplyr::select(genus,total,tame,aggr, unsel) %>% 
    tt()
```

```{r taxonomy_jitterplot_genus_location, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genus_summary %>%
  left_join(genome_metadata %>% dplyr::select(genus,phylum) %>% unique(),by="genus") %>%
  left_join(sample_metadata,by=join_by(sample==sample)) %>%
  filter(genus %in% genus_arrange[1:20]) %>%
  mutate(genus=factor(genus,levels=rev(genus_arrange[1:20]))) %>%
  filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=genus, group=genus, color=phylum, fill=phylum)) +
  scale_color_manual(values=phylum_colors[-8]) +
  scale_fill_manual(values=phylum_colors[-8]) +
  geom_boxplot(alpha=0.2)+
  geom_jitter(alpha=0.5) + 
  facet_grid(.~ gut_location)+
  theme_minimal() +
  theme(legend.position="none",
              strip.text.x = element_text(size = 14, color="black",face="bold"),
              axis.text.x = element_text(vjust = 0.5, size = 6),
              axis.text.y = element_text(size = 12),
              axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
              axis.title = element_text(size = 14, face = "bold"),
              axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
              axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)))+
  labs(y="Family", x="Relative abundance", color="Phylum")
```

```{r taxonomy_jitterplot_genus_origin, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genus_summary %>%
  left_join(genome_metadata %>% dplyr::select(genus,phylum) %>% unique(),by="genus") %>%
  filter(genus %in% genus_arrange[1:20]) %>%
  mutate(genus=factor(genus,levels=rev(genus_arrange[1:20]))) %>%
  filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=genus, group=genus, color=phylum, fill=phylum)) +
  scale_color_manual(values=phylum_colors[-8]) +
  scale_fill_manual(values=phylum_colors[-8]) +
  geom_boxplot(alpha=0.2)+
  geom_jitter(alpha=0.5) + 
  facet_grid(.~fox_behaviour)+
  theme_minimal() +
  theme(legend.position="none",
              strip.text.x = element_text(size = 14, color="black",face="bold"),
              axis.text.x = element_text(vjust = 0.5, size = 6),
              axis.text.y = element_text(size = 12),
              axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
              axis.title = element_text(size = 14, face = "bold"),
              axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
              axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)))+
  labs(y="Family", x="Relative abundance", color="Phylum")
```

## Core microbiota

```{r core_mag_upset, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
library(UpSetR)

genome_counts_rel <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>%
  column_to_rownames(., "genome")

#Presence/absence
genome_counts_rel_pa <- (genome_counts_rel > 0) * 1

#Add gut_location info
df <- as.data.frame(t(genome_counts_rel_pa)) %>%
  rownames_to_column("sample") %>%
  left_join(sample_metadata, by="sample") 

#Summarize by location
table_upset_analysis <- df %>%
  group_by(gut_location) %>%
  summarise(across(-sample, ~ as.integer(any(. > 0)))) %>%
  column_to_rownames("gut_location") %>%
  t() %>%
  as.data.frame()

location_colors <- c(
  F_colon = "#d6604d",
  M_colon = "#542788",
  D_ileum = "#fdb863",
  K_ileum = "#bb99d8"
)

#UpSet plot
location_upset_plot <- upset(table_upset_analysis,
      keep.order = TRUE,
      sets = rev(c("F_colon", "M_colon", "D_ileum", "K_ileum")),
      sets.bar.color = rev(location_colors),
      mb.ratio = c(0.55, 0.45),
      text.scale = c(2.5, 2.5, 2.5, 2.5) ,
      order.by = "freq")

location_upset_plot
```

```{r upset_plot_fox_behaviour, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
# Convert to relative abundance
genome_counts_rel <- genome_counts_filt %>%
  mutate_at(vars(-genome), ~ . / sum(.)) %>%
  column_to_rownames("genome")

# Presence/absence
genome_counts_rel_pa <- (genome_counts_rel > 0) * 1

# Add behaviour info
df <- as.data.frame(t(genome_counts_rel_pa)) %>%
  rownames_to_column("sample") %>%
  left_join(sample_metadata, by = "sample")

# Summarize by behaviour 
table_upset_analysis <- df %>%
  group_by(fox_behaviour) %>%
  summarise(across(-sample, ~ as.integer(any(. > 0)))) %>%
  column_to_rownames("fox_behaviour") %>%
  t() %>%
  as.data.frame()

# Behaviour colors
behaviour_colors <- c(
  tame  = "#1f77b4",  
  aggr  = "#d62728",  
  unsel = "#2ca02c"   
)

# UpSet plot
behaviour_upset_plot <- upset(table_upset_analysis,
      keep.order = TRUE,
      sets = rev(names(behaviour_colors)),       
      sets.bar.color = rev(unname(behaviour_colors)), 
      mb.ratio = c(0.55, 0.45),
      text.scale = c(2, 2, 2, 2.5) ,
      order.by = "freq")

behaviour_upset_plot
```



## Wilcoxon Test
H0 (null): The distribution of genome prevalence (presence) is the same across the two sample_type groups (gut tissue or gut content).
```{r core_mag_origin, warning=FALSE, comments="", message=FALSE}
mag_prevalence_location <- genome_counts_filt  %>%
  mutate_at(vars(-genome),~./sum(.)) %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  mutate(presence=ifelse(abundance>0,1,0)) %>% 
 dplyr::select(genome,sample,presence,gut_location, sample_type) %>%
  group_by(genome,gut_location, sample_type) %>%
  summarise(presence=ifelse(sum(presence)>0,1,0)) %>% 
  group_by(genome,sample_type) %>%
  summarise(presence=sum(presence))

mag_prevalence_location %>% 
  group_by(sample_type) %>%
  summarise(mean=mean(presence),sd=sd(presence)) %>%
  tt()

wilcox.test(presence ~ sample_type, data=mag_prevalence_location)  %>%
  tidy()
```

```{r core_mag_origin_boxplot, warning=FALSE, comments="", message=FALSE}
mag_prevalence_location %>% 
  ggplot(aes(x=sample_type,y=presence, color=sample_type, fill=sample_type)) +
    geom_boxplot(alpha = 0.5) +
    geom_point() +
    theme_minimal()
```
Comparing the prevalence/presence of the MAGs between aggressive and tame foxes
```{r core_mag_fox_behaviour, warning=FALSE, comments="", message=FALSE}
mag_prevalence_behaviour <- genome_counts_filt  %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #divide each column (except genome) by its sum to get relative abundance
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  filter(!grepl("unsel", fox_behaviour))%>% #to remove the unselected foxes
  mutate(presence=ifelse(abundance>0,1,0)) %>%  #converts abundance to presence/absence
 dplyr::select(genome,sample,presence,gut_location, fox_behaviour) %>%
  group_by(genome,gut_location, fox_behaviour) %>%
  summarise(presence=ifelse(sum(presence)>0,1,0)) %>%  #counts presence per combination of genome, gut location and fox behaviour
  group_by(genome,fox_behaviour) %>% 
  summarise(presence=sum(presence))  #sum the presences for each fox behaviour 

mag_prevalence_behaviour %>% 
  group_by(fox_behaviour) %>%
  summarise(mean=mean(presence),sd=sd(presence)) %>%
  tt()

wilcox.test(presence ~ fox_behaviour, data=mag_prevalence_behaviour)  %>%
  tidy()
```

```{r core_mag_behaviour_boxplot, warning=FALSE, comments="", message=FALSE}
behaviour_colors <- c( "#d62728", "#1f77b4","#2ca02c")

mag_prevalence_behaviour %>% 
  ggplot(aes(x=fox_behaviour,y=presence, color=fox_behaviour, fill=fox_behaviour)) +
    geom_boxplot() +
    geom_point() +
    scale_color_manual(values=behaviour_colors) +
    scale_fill_manual(values=str_c(behaviour_colors,"50")) +
    theme_minimal()
```

## Kruskal-Wallis
As we have >2 groups in gut location and fox behaviour, we will carry out Kruskal-Wallis tests to evaluate if there are significant differences between the groups in terms of MAG presence count (how many MAGs are present in each group). If the p-value < 0.05, we will do pairwise Wilcoxon tests, to evaluate which of the groups differ in this regard.

Gut location:
```{r kruskal wallis to test the gut locations}
mag_prevalence_gutlocation <- genome_counts_filt  %>%
  mutate_at(vars(-genome),~./sum(.)) %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>%
  mutate(presence=ifelse(abundance>0,1,0)) %>% 
 dplyr::select(genome,sample,presence,gut_location, fox_behaviour) %>%
  group_by(genome,gut_location) %>%
  summarise(presence=ifelse(sum(presence)>0,1,0)) 

# Kruskal–Wallis test (nonparametric ANOVA)
kruskal.test(presence ~ gut_location, data = mag_prevalence_gutlocation)
```

```{r pairwise wilcoxon tests}
pairwise.wilcox.test(mag_prevalence_gutlocation$presence,
                     mag_prevalence_gutlocation$gut_location,
                     p.adjust.method = "fdr")
```
```{r kruskal wallis to test the fox behaviours}
mag_prevalence_foxbehaviour <- genome_counts_filt  %>%
  mutate_at(vars(-genome),~./sum(.)) %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>%
  mutate(presence=ifelse(abundance>0,1,0)) %>% 
 dplyr::select(genome,sample,presence,gut_location, fox_behaviour) %>%
  group_by(genome,fox_behaviour) %>%
  summarise(presence=ifelse(sum(presence)>0,1,0)) 

#Kruskal–Wallis test (nonparametric ANOVA)
kruskal.test(presence ~ fox_behaviour, data = mag_prevalence_foxbehaviour)
```
```{r pairwise wilcoxon tests fox behaviour}
pairwise.wilcox.test(mag_prevalence_foxbehaviour$presence,
                     mag_prevalence_foxbehaviour$fox_behaviour,
                     p.adjust.method = "fdr")
```







