---
title: "AlberdiLab | Silver fox metagenomics"
subtitle: MSc project
author:
  - Irene Martínez, M Thomas P Gilbert, Antton Alberdi^[University of Copenhagen, antton.alberdi@sund.ku.dk]
date: "Last update: `r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
url: https://alberdilab.github.io/silverfox_metagenomics
description: |
  Data analysis code for the study of Belyayev silverfox metagenomes
link-citations: yes
github-repo: alberdilab/silverfox_metagenomics
---

```{r knitr_opts, echo=FALSE}
knitr::opts_chunk$set(
    class.source = "script-source",
    class.output = "script-output",
    comment = NA)
```

# Introduction

This webbook contains all the code used for data analysis in study of gut microbiomes of Belyayev tame and aggressive silverfoxes.

## Prepare the R environment

### Environment

To reproduce all the analyses locally, clone this repository in your computer using:

```
RStudio > New Project > Version Control > Git
```

And indicating the following git repository:

> https://github.com/alberdilab/silverfox_metagenomics.git

Once the R project has been created, follow the instructions and code chunks shown in this webbook.

### Libraries

The following R packages are required for the data analysis.

```{r load_libraries, warning=FALSE, comments="", message=FALSE}
# Base
library(R.utils)
library(knitr)
library(tidyverse)
library(devtools)
library(tinytable)
library(rairtable)
library(janitor)
library(broom)

# For tree handling
library(ape)
library(phyloseq)
library(phytools)

# For plotting
library(ggplot2)
library(ggrepel)
library(ggpubr)
library(ggnewscale)
library(gridExtra)
library(ggtreeExtra)
library(ggtree)
library(ggh4x)
library(UpSetR)
library(viridis)

# For statistics
library(spaa)
library(vegan)
library(Rtsne)
library(geiger)
library(hilldiv2)
library(distillR)
library(ANCOMBC)
library(lme4)
library(nlme)
library(pairwiseAdonis)
library(emmeans)
library(pheatmap)
library(rstatix)
library(uwot)
```

<!--chapter:end:index.Rmd-->

# Data preparation

## Load data

Load the original data files outputted by the bioinformatic pipeline.

### Sample metadata

```{r load_sample_metadata, warning=FALSE, comments="", message=FALSE, eval=FALSE}
sample_metadata <- read_csv("data/sample_metadata.csv") %>%
  rename(sample =1, 
         gut_location = 2,
         fox_behaviour = 3, 
         fox_sex = 4,
         sample_type = 5, 
         year = 6)
```


### Read counts

```{r load_read_counts, warning=FALSE, comments="", message=FALSE, eval=FALSE}
read_counts <- read_tsv("data/counts.tsv") %>%
    rename(genome=1)%>%
    select(any_of(c("genome", sample_metadata$sample)))

```


### Filter read counts

```{r filter_read_counts, warning=FALSE, comments="", message=FALSE, eval=FALSE}
read_counts_filt <- read_counts %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  select(one_of(c("genome",sample_metadata$sample)))

```


### Genome info

```{r load_genome_info, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_info <- read_csv("data/genomeInfo.csv") %>%
   mutate(genome = gsub(".fa$", "", genome)) %>%
   semi_join(., read_counts, by = "genome") %>%
   arrange(match(genome,read_counts$genome))
```

### Genome taxonomy

```{r load_genome_taxonomy, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_taxonomy <- read_tsv("data/genome_taxonomy.tsv") %>%
  rename(genome = 1)

genome_taxonomy_expanded <- genome_taxonomy %>% 
  separate(classification,
           into = c("domain", "phylum", "class", "order", "family", "genus", "species"),
           sep = ";",
           fill = "right", remove = FALSE) %>%
  mutate(across(domain:species, ~ str_replace(.x, "^[a-z]__", "")))%>%
  mutate(phylum = case_when(
    phylum == "Actinobacteriota" ~ "Actinomycetota",
    phylum == "Firmicutes" ~ "Bacillota",
    phylum == "Firmicutes_A" ~ "Bacillota_A",
    phylum == "Firmicutes_C" ~ "Bacillota_C",
    phylum == "Cyanobacteria" ~ "Cyanobacteriota",
    phylum == "Proteobacteria" ~ "Pseudomonadota",
    TRUE ~ phylum))
```

### Genome metadata
```{r load_genome_metadata, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_metadata <- genome_taxonomy_expanded %>%
  left_join(genome_info, by = "genome")
```


### Genome coverage

```{r load_bases, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_coverage <- read_tsv("data/bases.tsv") %>%
    rename(genome=1) %>%
    select(one_of(c("genome",sample_metadata$sample))) %>%
    left_join(genome_info %>% select(genome, length), by = "genome") %>%
    mutate(across(where(is.numeric), ~ .x / length))
```


### Genome tree

```{r load_genome_tree, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_tree <- read_tree("data/gtdbtk.backbone.bac120.classify.tree")
genome_tree$tip.label <- str_replace_all(genome_tree$tip.label,"'", "") #remove single quotes in MAG names
genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome) # keep only MAG tips
```
### Genome annotations

```{r load_gene_annotations, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_annotations <- read_tsv("data/gene_annotations.tsv.xz") %>%
  mutate(genome = sub("\\^.*", "", gene)) %>%
  select(1, genome, everything())
```

## Create working objects

Transform the original data files into working objects for downstream analyses.

### Filter reads by coverage

```{r filter_coverage, warning=FALSE, comments="", message=FALSE, eval=FALSE}
min_coverage=0.3
read_counts_filt <- genome_coverage %>%
  select(-length) %>% 
  mutate(across(where(is.numeric), ~ ifelse(. > min_coverage, 1, 0))) %>%
  mutate(across(-1, ~ . * read_counts[[cur_column()]])) 
```


### Transform reads into genome counts

```{r calculate_genome_counts_unfiltered, warning=FALSE, comments="", message=FALSE, eval=FALSE}
readlength=150
genome_counts <- read_counts %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

```{r calculate_genome_counts_filtered, warning=FALSE, comments="", message=FALSE, eval=FALSE}
readlength=150
genome_counts_filt <- read_counts_filt %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

### Distill annotations into GIFTs 

```{r distill_annotations, warning=FALSE, comments="", message=FALSE, eval=FALSE}
genome_gifts <- distill(genome_annotations,GIFT_db,genomecol= 2, annotcol=c(6,7,8,9), verbosity = F)
```

## Prepare color scheme

[AlberdiLab](www.alberdilab.dk) projects use unified color schemes developed for the [Earth Hologenome Initiative](www.earthhologenome.org), to facilitate figure interpretation.

```{r get_ehi_colors, warning=FALSE, comments="", message=FALSE, eval=FALSE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    mutate(phylum=str_remove_all(phylum, "p__")) %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    dplyr::select(phylum, colors) %>% 
    unique() %>%
    arrange(phylum) %>%
    pull(colors, name=phylum)

location_colors <- c('#9467bd','#7b3294','#ff7f0e','#e66101','#BFA366','#6E5244')

behaviour_colors <- c(
  tame = "#1f77b4",   
  aggr = "#d62728",   
  unsel = "#2ca02c"   
)
```

## Wrap working objects

All working objects are wrapped into a single Rdata object to facilitate downstream usage.

```{r wrap_working_objects, warning=FALSE, comments="", message=FALSE, eval=FALSE}
save(sample_metadata, 
     genome_metadata, 
     read_counts, 
     genome_annotations,
     genome_counts, 
     genome_counts_filt, 
     genome_tree,
     genome_gifts, 
     phylum_colors,
     location_colors,
     behaviour_colors,
     file = "data/data.Rdata")
```


<!--chapter:end:01_data_preparation.Rmd-->

# Data statistics

## Sequencing reads statistics

```{r load_data_stats}
load("data/data.Rdata")
```

### Loading data

```{r load preprocessing data}
preprocessing <- read_tsv("data/preprocessing.tsv")
```

```{r reads_stats}
preprocessing %>% 
    summarise(Total=sum(reads_metagenomic * 150 / 1000000000) %>% round(2), 
              mean=mean(reads_metagenomic * 150 / 1000000000) %>% round(2),
              sd=sd(reads_metagenomic * 150 / 1000000000) %>% round(2)) %>%
    unite("Average",mean, sd, sep = " ± ", remove = TRUE) %>%
    tt()
```

### DNA fractions

```{r dna_fractions_stats}
sequence_fractions <- read_counts %>%
  pivot_longer(-genome, names_to = "sample", values_to = "value") %>%
  group_by(sample) %>%
  summarise(mags = sum(value)) %>%
	left_join(preprocessing, by = join_by(sample == sample)) %>%
	select(sample,mags,bases_metagenomic,bases_host ,bases_discarded, bases_raw) %>%
	mutate(mags_bases = mags*146) %>%
  mutate(bases_discarded_proportion = bases_discarded/bases_raw) %>%
	mutate(lowqual_bases = ((bases_metagenomic+bases_host)/(1-bases_discarded_proportion))-(bases_metagenomic+bases_host)) %>%
	mutate(unmapped_bases = bases_metagenomic - mags_bases) %>%
	mutate(unmapped_bases = ifelse(unmapped_bases < 0, 0, unmapped_bases)) %>%
	select(sample, lowqual_bases, bases_host, unmapped_bases, mags_bases)

sequence_fractions %>%
  mutate_at(vars(-sample), ~./1000000000) %>%
  rename("Sample"=1, "Low quality"=2, "Mapped to host"=3, "Unmapped"=4, "Mapped to MAGs"=5) %>%
  tt()

#mean
sequence_fractions %>%
  mutate_at(vars(-sample), ~./1000000000) %>%
  rename("Sample"=1, "Low quality"=2, "Mapped to host"=3, "Unmapped"=4, "Mapped to MAGs"=5) %>%
  summarise(across(where(is.numeric), mean, na.rm = TRUE)) %>%
  tt()
#SD
sequence_fractions %>%
  mutate_at(vars(-sample), ~./1000000000) %>%
  rename("Sample"=1, "Low quality"=2, "Mapped to host"=3, "Unmapped"=4, "Mapped to MAGs"=5) %>%
  summarise(across(where(is.numeric), sd, na.rm = TRUE)) %>% 
  tt()

```


```{r dna_fractions_plot, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
sequence_fractions_plot <- sequence_fractions %>%
	pivot_longer(!sample, names_to = "fraction", values_to = "value") %>%
	mutate(value = value / 1000000000) %>%
	mutate(fraction = factor(fraction, levels = c("lowqual_bases","bases_host","unmapped_bases","mags_bases"))) 
  
ggplot(sequence_fractions_plot, aes(x = sample, y = value, fill=fraction)) +
    geom_bar(position="stack", stat = "identity") +
    scale_fill_manual(name="Sequence type",
                  breaks=c("lowqual_bases","bases_host","unmapped_bases","mags_bases"),
                  labels=c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                  values=c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c"))+
    labs(x = "Samples", y = "Amount of data (GB)") +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size=6),legend.position = "bottom")
```


```{r dna_fractions_plot_sample_type, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
sequence_fractions_type <- sequence_fractions_plot %>%
  left_join(sample_metadata) 

ggplot(sequence_fractions_type, aes(x = sample, y = value, fill = fraction)) +
    geom_bar(position = "stack", stat = "identity") +
    scale_fill_manual(name = "Sequence type",
                      breaks = c("lowqual_bases","bases_host","unmapped_bases","mags_bases"),
                      labels = c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                      values = c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c")) +
    labs(x = "Samples", y = "Amount of data (GB)") +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 6),
          legend.position = "bottom") +
    facet_wrap(~ sample_type, scales = "free_x", nrow = 1)
    
```



```{r dna_fractions_plot_gut_location, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(sequence_fractions_type, aes(x = sample, y = value, fill = fraction)) +
    geom_bar(position = "stack", stat = "identity") +
    scale_fill_manual(name = "Sequence type",
                      breaks = c("lowqual_bases","bases_host","unmapped_bases","mags_bases"),
                      labels = c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                      values = c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c")) +
    labs(x = "Samples", y = "Amount of data (GB)") +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 6),
          legend.position = "bottom") +
    facet_wrap(~ gut_location, scales = "free_x", nrow = 1)
    
```


```{r dna_fractions_plot_sample_types, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(sequence_fractions_type, aes(x = sample, y = value, fill = fraction)) +
    geom_bar(position = "stack", stat = "identity") +
    scale_fill_manual(name = "Sequence type",
                      breaks = c("lowqual_bases","bases_host","unmapped_bases","mags_bases"),
                      labels = c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                      values = c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c")) +
       labs(x = "Samples", y = "Amount of data (GB)") +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 6),
          legend.position = "bottom") +
    facet_nested(~ sample_type + gut_location,
                 scales = "free_x", space = "free_x")

```



```{r dna_fractions_plot_fox_behaviour, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
ggplot(sequence_fractions_type, aes(x = sample, y = value, fill = fraction)) +
    geom_bar(position = "stack", stat = "identity") +
    scale_fill_manual(name = "Sequence type",
                      breaks = c("lowqual_bases","bases_host","unmapped_bases","mags_bases"),
                      labels = c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                      values = c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c")) +
    labs(x = "Samples", y = "Amount of data (GB)") +
    theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 6),
          legend.position = "bottom") +
    facet_wrap(~ fox_behaviour, scales = "free_x", nrow = 1)
    
```


<!--chapter:end:02_data_statistics.Rmd-->

# MAG catalogue

```{r load_data_mag, message=FALSE, warning=FALSE, echo=FALSE}
load("data/data.Rdata")
```
## Genome phylogeny

Circular tree with genome size and completeness
```{r genome_phylogeny_completeness, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
# Generate the phylum color heatmap
phylum_heatmap <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(genome,phylum) %>%
    mutate(phylum = factor(phylum, levels = unique(phylum))) %>%
    column_to_rownames(var = "genome")


# Generate  basal tree
circular_tree <- force.ultrametric(genome_tree, method="extend") %>% # extend to ultrametric for the sake of visualisation
    ggtree(., layout="fan", open.angle=10, size=0.5)


# Add phylum ring
circular_tree <- gheatmap(circular_tree, phylum_heatmap, offset=0.05, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=phylum_colors) +
  labs(fill="Phylum")
        #theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))

# Flush color scale to enable a new color scheme in the next ring
circular_tree <- circular_tree + new_scale_fill()

# Add completeness ring
circular_tree <- circular_tree +
        new_scale_fill() +
        scale_fill_gradient(low = "#d1f4ba", high = "#f4baba") +
        geom_fruit(
                data=genome_metadata,
                geom=geom_bar,
                mapping = aes(x=completeness, y=genome, fill=contamination),
                offset = 0.24,
                pwidth = 0.1,
                orientation="y",
              stat="identity")+
  labs(fill="Contamination")

# Add genome-size ring
circular_tree <-  circular_tree + new_scale_fill()

circular_tree <-  circular_tree +
  geom_fruit(
    data=genome_metadata,
    geom=geom_bar,
    mapping = aes(x=length, y=genome),
    fill = "#1e6e55",
    offset = 0.05,
    orientation="y",
    stat="identity")


# Add text
circular_tree <-  circular_tree +
        annotate('text', x=2.15, y=0, label='            Phylum', family='arial', size=3.5) +
        annotate('text', x=2.6, y=0, label='                         Genome quality', family='arial', size=3.5) +
        annotate('text', x=2.8, y=0, label='                     Genome size', family='arial', size=3.5) 

#Plot circular tree
circular_tree %>% open_tree(30) %>% rotate_tree(90)
```

Investigating large genome MAG:
```{r large_genome}
large_genome_mag <- genome_metadata %>% 
  arrange(desc(size)) %>% 
  mutate(size_Mb = size /1000000)%>%
  select(genus, size_Mb ,msa_percent, completeness, contamination, score) 
  

head(large_genome_mag)
```
Paraclostridium species tend to have genome sizes of 2.9-3.6Mb. This MAG seems to have lower msa_percent, is this relevant?




Circular tree with genome presence/absence in different sample locations and fox behaviour
```{r genome_phylogeny, message=FALSE, warning=FALSE}
# Generate the phylum color heatmap
phylum_heatmap <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    mutate(phylum=str_remove_all(phylum, "p__")) %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    dplyr::select(genome,phylum) %>%
    mutate(phylum = factor(phylum, levels = unique(phylum))) %>%
    column_to_rownames(var = "genome")

# Generate prevalence data
prevalence_data <- genome_counts_filt  %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  group_by(genome,gut_location) %>% 
  summarise(presence=ifelse(sum(abundance)>0,1,0)) %>% 
  group_by(genome) %>%
  summarise(prevalence=sum(presence))

# Generate F_colon heatmap
f_colon_heatmap <- genome_counts_filt  %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  filter(gut_location =="F_colon") %>% 
  group_by(genome) %>% 
  summarise(presence=ifelse(sum(abundance)>0,"present","absent")) %>% 
  column_to_rownames(var="genome")

# Generate M_colon heatmap
m_colon_heatmap <- genome_counts_filt  %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  filter(gut_location =="M_colon") %>% 
  group_by(genome) %>% 
  summarise(presence=ifelse(sum(abundance)>0,"present","absent")) %>% 
  column_to_rownames(var="genome")

# Generate D_ileum heatmap
d_ileum_heatmap <- genome_counts_filt  %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  filter(gut_location =="D_ileum") %>% 
  group_by(genome) %>% 
  summarise(presence=ifelse(sum(abundance)>0,"present","absent")) %>% 
  column_to_rownames(var="genome")

# Generate K_ileum heatmap
k_ileum_heatmap <- genome_counts_filt  %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  filter(gut_location =="K_ileum") %>% 
  group_by(genome) %>% 
  summarise(presence=ifelse(sum(abundance)>0,"present","absent")) %>% 
  column_to_rownames(var="genome")

# Generate Gut content heatmap
gut_content_heatmap <- genome_counts_filt  %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  filter(sample_type =="Gut content") %>% 
  group_by(genome) %>% 
  summarise(presence=ifelse(sum(abundance)>0,"present","absent")) %>% 
  column_to_rownames(var="genome")


# Generate Gut tissue heatmap
gut_tissue_heatmap <- genome_counts_filt  %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  filter(sample_type =="Gut tissue") %>% 
  group_by(genome) %>% 
  summarise(presence=ifelse(sum(abundance)>0,"present","absent")) %>% 
  column_to_rownames(var="genome")

# Generate tame heatmap
tame_heatmap <- genome_counts_filt  %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  filter(fox_behaviour=="tame") %>% 
  group_by(genome) %>% 
  summarise(presence=ifelse(sum(abundance)>0,"present","absent")) %>% 
  column_to_rownames(var="genome")

# Generate aggressive heatmap
aggressive_heatmap <- genome_counts_filt  %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  filter(fox_behaviour=="aggr") %>% 
  group_by(genome) %>% 
  summarise(presence=ifelse(sum(abundance)>0,"present","absent")) %>% 
  column_to_rownames(var="genome")

# Generate unselected heatmap
unsel_heatmap <- genome_counts_filt  %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  filter(fox_behaviour=="unsel") %>% 
  group_by(genome) %>% 
  summarise(presence=ifelse(sum(abundance)>0,"present","absent")) %>% 
  column_to_rownames(var="genome")
```

```{r genome_phylogeny_plot, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
# Generate  basal tree
circular_tree <- force.ultrametric(genome_tree, method="extend") %>% # extend to ultrametric for the sake of visualisation
    ggtree(., layout="fan", open.angle=10, size=0.3)

# Add phylum ring
circular_tree <- gheatmap(circular_tree, phylum_heatmap, offset=0, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=phylum_colors) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0)) +
        new_scale_fill()

# Add F_colon ring
circular_tree <- gheatmap(circular_tree, f_colon_heatmap, offset=0.2, width=0.05, colnames=FALSE) +
        scale_fill_manual(values = c("absent" = "#ffffff", "present" = "#d6604d")) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0)) +
        new_scale_fill()

# Add M_colon ring
circular_tree <- gheatmap(circular_tree, m_colon_heatmap, offset=0.4, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=c("#ffffff","#542788")) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0)) +
        new_scale_fill()

# Add D_ileum ring
circular_tree <- gheatmap(circular_tree, d_ileum_heatmap, offset=0.3, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=c("#ffffff","#fdb863")) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0)) +
        new_scale_fill()

# Add K_ileum ring
circular_tree <- gheatmap(circular_tree, k_ileum_heatmap, offset=0.5, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=c("absent" =  "#ffffff", "present" = "#bb99d8")) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0)) +
        new_scale_fill()



# Add prevalence ring
circular_tree <-  circular_tree +
        new_scale_fill() +
        scale_fill_manual(values = "#cccccc") +
        geom_fruit(
             data=prevalence_data,
             geom=geom_bar,
             mapping = aes(x=prevalence, y=genome),
                 offset = 0.4,
                 orientation="y",
         stat="identity")

# Add tame ring
circular_tree <- gheatmap(circular_tree, tame_heatmap, offset=1.2, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=c("#ffffff","#1f77b4")) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0)) +
        new_scale_fill()

# Add aggressive ring
circular_tree <- gheatmap(circular_tree, aggressive_heatmap, offset=1.3, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=c("#ffffff","#d62728")) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0)) +
        new_scale_fill()

# Add unselected ring
circular_tree <- gheatmap(circular_tree, unsel_heatmap, offset=1.4, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=c("#ffffff","#2ca02c")) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0)) +
        new_scale_fill()


# Add text
circular_tree <-  circular_tree +
        annotate('text', x=2.05, y= 0, label='Phylum', family='arial', size=3.5) +
        annotate('text', x=2.4, y=0, label='Gut location', family='arial', size=3.5) +
        annotate('text', x=3.1, y=0, label='Prevalence', family='arial', size=3.5) +
        annotate('text', x= 3.5,y=0, label='Fox behaviour', family='arial', size=3.5) 


#Plot circular tree
circular_tree %>% open_tree(30) %>% rotate_tree(90)
```

### Taxonomy overview

```{r genome_taxonomy_bacteria}
tax_mag <-genome_metadata %>%
  group_by(phylum) %>%
  summarise(mag_n=n())

tax_mag %>%
  mutate(percetage_mag=round(mag_n*100/sum(mag_n), 2)) %>%
  arrange(-percetage_mag) %>%
  tt()
```

### Mag size (MB)

```{r mag size}
genome_metadata <- genome_metadata %>%
  mutate(corrected_size=100*length/completeness) %>%
  arrange(completeness)
```

```{r mag_size_all_mean, message=FALSE, warning=FALSE}
genome_metadata %>%
  summarise(Average_corrected_size=mean(corrected_size))
```

## Genome quality

```{r genome_quality}
tibble(Completeness=
         paste0(round(genome_metadata$completeness %>% mean(),2),
                "±",
                round(genome_metadata$completeness %>% sd(),2)),
       Contamination=
           paste0(round(genome_metadata$contamination %>% mean(),2),
                "±",
                round(genome_metadata$contamination %>% sd(),2))) %>%
  tt()
```


```{r genome_quality_plot, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}

#Generate quality biplot
genome_biplot <- genome_metadata %>%
  dplyr::select(c(genome,phylum,completeness,contamination,length)) %>%
  arrange(match(genome, rev(genome_tree$tip.label))) %>% #sort MAGs according to phylogenetic tree
  ggplot(aes(x=completeness,y=contamination,size=length,color=phylum)) +
              geom_point(alpha=0.7) +
                    xlim(c(70,100)) +
                    ylim(c(10,0)) +
                    scale_color_manual(values=phylum_colors) +
                    labs(y= "Contamination", x = "Completeness") +
                    theme_classic() +
                    theme(legend.position = "none")

#Generate contamination boxplot
genome_contamination <- genome_metadata %>%
            ggplot(aes(y=contamination)) +
                    ylim(c(10,0)) +
                    geom_boxplot(colour = "#999999", fill="#cccccc") +
                    theme_classic() +
                    theme(legend.position = "none",
                    axis.line = element_blank(),
                    axis.title = element_blank(),
                    axis.text=element_blank(),
                    axis.ticks=element_blank(),
                        plot.margin = unit(c(0, 0, 0.40, 0),"inches")) #add bottom-margin (top, right, bottom, left)

#Generate completeness boxplot
genome_completeness <- genome_metadata %>%
        ggplot(aes(x=completeness)) +
                xlim(c(70,100)) +
                geom_boxplot(colour = "#999999", fill="#cccccc") +
                theme_classic() +
                theme(legend.position = "none",
                    axis.line = element_blank(),
                    axis.title = element_blank(),
                    axis.text=element_blank(),
                    axis.ticks=element_blank(),
                    plot.margin = unit(c(0, 0, 0, 0.50),"inches")) #add left-margin (top, right, bottom, left)

#Render composite figure
#pdf("figures/completeness_contamination.pdf",width=10, height=5)
grid.arrange(grobs = list(genome_completeness,genome_biplot,genome_contamination),
        layout_matrix = rbind(c(1,1,1,1,1,1,1,1,1,1,1,4),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3)))
#dev.off()
```


## Functional overview

#### Predicted genes
```{r predicted_stats, message=FALSE, warning=FALSE}
pred_genes <- genome_annotations %>%
  nrow()

cat(pred_genes)
```

#### Number of annotated genes and percentages
```{r annotation_stats, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
#How many genes have at least 1 annotation
genome_annota <- genome_annotations %>%
  filter(if_any(c(kegg, pfam, cazy), ~ !is.na(.))) %>%
  nrow()

cat(genome_annota)

#Percentage of predicted genes with at least 1 annotation
genome_annota*100/pred_genes
```

#### Number of KEGG annotatated genes and percentages
```{r kegg_stats_gut, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
# KEGG annotation
kegg_annota <- genome_annotations %>%
  filter(!is.na(kegg)) %>%
  nrow()
cat(kegg_annota)

# KEGG annotation percentage
kegg_annota*100/genome_annota
```

#### Function tree
```{r function_heatmap, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}

# Aggregate basal GIFT into elements
function_table <- genome_gifts %>%
    to.elements(., GIFT_db)

# Generate  basal tree
function_tree <- force.ultrametric(genome_tree, method="extend") %>%
                ggtree(., size = 0.3) 

#Add phylum colors next to the tree tips
function_tree <- gheatmap(function_tree, phylum_heatmap, offset=0, width=0.1, colnames=FALSE) +
            scale_fill_manual(values=phylum_colors) +
            labs(fill="Phylum")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

#Add functions heatmap
function_tree <- gheatmap(function_tree, function_table, offset=0.5, width=3.5, colnames=FALSE) +
            vexpand(.08) +
            coord_cartesian(clip = "off") +
            scale_fill_gradient(low = "#f4f4f4", high = "steelblue", na.value="white") +
            labs(fill="GIFT")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

# Add completeness barplots
function_tree <- function_tree +
            geom_fruit(data=genome_metadata,
            geom=geom_bar,
            grid.params=list(axis="x", text.size=2, nbreak = 1),
            axis.params=list(vline=TRUE),
            mapping = aes(x=length, y=genome, fill=completeness),
                 offset = 3.8,
                 orientation="y",
                 stat="identity") +
            scale_fill_gradient(low = "#cf8888", high = "#a2cc87") +
            labs(fill="Genome\ncompleteness")

function_tree
```

## Functional ordination

PCoA with Euclidean distances:
```{r pcoa_ordination, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
gift_pcoa <- genome_gifts %>%
    to.elements(., GIFT_db) %>%
    as.data.frame() %>%
    vegdist(method="euclidean") %>%
    pcoa()

gift_pcoa_rel_eigen <- gift_pcoa$values$Relative_eig[1:10]


# Get genome positions
gift_pcoa_vectors <- gift_pcoa$vectors %>% #extract vectors
  as.data.frame() %>% 
  select(Axis.1,Axis.2) # keep the first 2 axes

gift_pcoa_eigenvalues <- gift_pcoa$values$Eigenvalues[c(1,2)]


#For the black arrows: Functional group loadings
gift_pcoa_gifts <- cov(genome_gifts, scale(gift_pcoa_vectors)) %*%   diag((gift_pcoa_eigenvalues/(nrow(genome_gifts)-1))^(-0.5)) %>%
  as.data.frame() %>% 
  rename(Axis.1=1,Axis.2=2) %>% 
  rownames_to_column(var="label") %>% 
  #get function summary vectors
  mutate(func=substr(label,1,3)) %>% 
  group_by(func) %>% 
  summarise(Axis.1=mean(Axis.1),
            Axis.2=mean(Axis.2)) %>% 
  rename(label=func) %>% 
  filter(!label %in% c("S01","S02","S03"))
```

```{r pcoa_ordination_plot, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
set.seed(101)
scale <- 20 # scale for vector loadings (to make arrows visible)
gift_pcoa_vectors %>% 
  rownames_to_column(var="genome") %>% 
  left_join(genome_metadata, by="genome") %>%
  ggplot() +
      #genome positions
      scale_color_manual(values=phylum_colors)+
      geom_point(aes(x=Axis.1,y=Axis.2, color=phylum, size=length), 
                 alpha=0.9, shape=16) +
      #scale_color_manual(values=phylum_colors) +
      scale_size_continuous(range = c(0.1,5)) +
      #loading positions
      geom_segment(data=gift_pcoa_gifts, 
                   aes(x=0, y=0, xend=Axis.1 * scale, yend=Axis.2 * scale),
                    arrow = arrow(length = unit(0.3, "cm"), 
                    type = "open", 
                    angle = 25),
                    linewidth = 0.5, 
                    color = "black") +
     #Primary and secondary scale adjustments
     scale_x_continuous(name = paste0("PCoA1 (",round(gift_pcoa_rel_eigen[1]*100, digits = 2), " %)"),
                      sec.axis = sec_axis(~ . / scale, name = "Loadings on PCoA1")
            ) +
     scale_y_continuous(name = paste0("PCoA2 (",round(gift_pcoa_rel_eigen[2]*100, digits = 2), " %)"),
                      sec.axis = sec_axis(~ . / scale, name = "Loadings on PCoA2")
            ) +
    geom_label_repel(data = gift_pcoa_gifts,
                     aes(label = label, x = Axis.1 * scale, y = Axis.2 * scale),
                     segment.color = 'transparent') +
    xlim(-3,3) + 
    ylim(-2.5,2.5) +
    theme_minimal() + 
    theme(legend.position = "none")+ 
  labs(
    x = paste0("PCoA1 (", round(gift_pcoa_rel_eigen[1]*100, 2), " %)"),
    y = paste0("PCoA2 (", round(gift_pcoa_rel_eigen[2]*100, 2), " %)")
  )
```

```{r arrow functions euclidean}
arrow_functions <- gift_pcoa_gifts %>%
  left_join(GIFT_db, by = c("label" = "Code_function")) %>%
  group_by(label) %>%
  summarise(
    Axis.1 = mean(Axis.1),
    Axis.2 = mean(Axis.2),
    Function = first(Function),
    Domain  = first(Domain),
    Length  = sqrt(Axis.1^2 + Axis.2^2), # vector length
    .groups = "drop"
  ) %>%
  arrange(desc(Length))  # sort so most important arrows are first

arrow_functions


```

PCoA with Bray-Curtis (relative composition)
```{r pcoa_ordination_bray, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
gift_pcoa_bray <- genome_gifts %>%
    to.elements(., GIFT_db) %>%
    as.data.frame() %>%
    vegdist(method="bray") %>%
    pcoa()

gift_pcoa_rel_eigen_bray <- gift_pcoa_bray$values$Relative_eig[1:10]


# Get genome positions
gift_pcoa_vectors_bray <- gift_pcoa_bray$vectors %>% #extract vectors
  as.data.frame() %>% 
  select(Axis.1,Axis.2) # keep the first 2 axes

gift_pcoa_eigenvalues_bray <- gift_pcoa_bray$values$Eigenvalues[c(1,2)]

gift_pcoa_gifts_bray <- cov(genome_gifts, scale(gift_pcoa_vectors_bray)) %*% diag((gift_pcoa_eigenvalues_bray/(nrow(genome_gifts)-1))^(-0.5)) %>%
  as.data.frame() %>% 
  rename(Axis.1=1,Axis.2=2) %>% 
  rownames_to_column(var="label") %>% 
  #get function summary vectors
  mutate(func=substr(label,1,3)) %>% 
  group_by(func) %>% 
  summarise(Axis.1=mean(Axis.1),
            Axis.2=mean(Axis.2)) %>% 
  rename(label=func) %>% 
  filter(!label %in% c("S01","S02","S03"))

```

```{r pcoa_ordination_plot_bray, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
set.seed(101)
scale <- 20 # scale for vector loadings
gift_pcoa_vectors_bray %>% 
  rownames_to_column(var="genome") %>% 
  left_join(genome_metadata, by="genome") %>%
  ggplot() +
      #genome positions
      scale_color_manual(values=phylum_colors)+
      geom_point(aes(x=Axis.1,y=Axis.2, color=phylum, size=length), 
                 alpha=0.9, shape=16) +
      #scale_color_manual(values=phylum_colors) +
      scale_size_continuous(range = c(0.1,5)) +
      #loading positions
      geom_segment(data=gift_pcoa_gifts_bray, 
                   aes(x=0, y=0, xend=Axis.1 * scale, yend=Axis.2 * scale),
                    arrow = arrow(length = unit(0.3, "cm"), 
                    type = "open", 
                    angle = 25),
                    linewidth = 0.5, 
                    color = "black") +
     #Primary and secondary scale adjustments
     scale_x_continuous(name = paste0("PCoA1 (",round(gift_pcoa_rel_eigen_bray[1]*100, digits = 2), " %)"),
                      sec.axis = sec_axis(~ . / scale, name = "Loadings on PCoA1")
            ) +
     scale_y_continuous(name = paste0("PCoA2 (",round(gift_pcoa_rel_eigen_bray[2]*100, digits = 2), " %)"),
                      sec.axis = sec_axis(~ . / scale, name = "Loadings on PCoA2")
            ) +
    geom_label_repel(data = gift_pcoa_gifts_bray,
                     aes(label = label, x = Axis.1 * scale, y = Axis.2 * scale),
                     segment.color = 'transparent') +
   xlim(-0.8,0.8) + 
    ylim(-0.5,0.5) +
    theme_minimal() + 
    theme(legend.position = "none")+ 
  labs(
    x = paste0("PCoA1 (", round(gift_pcoa_rel_eigen[1]*100, 2), " %)"),
    y = paste0("PCoA2 (", round(gift_pcoa_rel_eigen[2]*100, 2), " %)")
  )
```

```{r arrow functions bray}
arrow_functions_bray <- gift_pcoa_gifts_bray %>%
  left_join(GIFT_db, by = c("label" = "Code_function")) %>%
  group_by(label) %>%
  summarise(
    Axis.1 = mean(Axis.1),
    Axis.2 = mean(Axis.2),
    Function = first(Function),
    Domain  = first(Domain),
    Length  = sqrt(Axis.1^2 + Axis.2^2),
    .groups = "drop"
  ) %>%
  arrange(desc(Length))

arrow_functions_bray

```



UMAP

```{r umap_parameter_grid, message=FALSE, fig.height=12, fig.width=15}
set.seed(1001)

#parameters to test
neighbors_values <- c(5, 15, 30)
min_dist_values <- c(0.1, 0.3, 0.5)

umap_plots <- list()

# Loop over combinations
for (nn in neighbors_values) {
  for (md in min_dist_values) {
    
    umap_result <- umap(function_table,
                        n_neighbors = nn,
                        min_dist = md,
                        metric = "euclidean")
    
    df_umap <- as.data.frame(umap_result) %>%
      mutate(genome = rownames(function_table)) %>%
      inner_join(genome_metadata, by="genome") %>%
      rename(UMAP1 = V1, UMAP2 = V2)
    
    p <- ggplot(df_umap, aes(x = UMAP1, y = UMAP2,
                             color = phylum, size = length)) +
      geom_point(alpha = 0.7, shape=16) +
      scale_color_manual(values = phylum_colors) +
      theme_minimal() +
      labs(title = paste0("UMAP (neighbors=", nn,
                          ", min_dist=", md, ")"),
           color="Phylum", size="Genome size") +
      guides(color = guide_legend(override.aes = list(size = 5)))
    
    p <- p + theme(legend.position = "none")
    umap_plots[[paste(nn, md, sep="_")]] <- p
    
  }
}
ggpubr::ggarrange(plotlist = umap_plots, ncol = 3, nrow = 3)


```

```{r testing_perplexity}
set.seed(1001)

# Trying different perplexity values
perplexities <- c(5, 7,8, 9, 10, 11, 12, 13, 15)

plots <- lapply(perplexities, function(p) {
  tsne_result <- Rtsne(
    X = function_table, 
    dims = 2, 
    perplexity = p, 
    check_duplicates = FALSE
  )
  
  tsne_df <- as.data.frame(tsne_result$Y) %>%
    mutate(genome = rownames(function_table)) %>%
    inner_join(genome_metadata, by = "genome") %>%
    rename(tSNE1 = "V1", tSNE2 = "V2")
  
  ggplot(tsne_df, aes(x = tSNE1, y = tSNE2, color = phylum)) +
    geom_point(alpha = 0.7, size = 2) +
    scale_color_manual(values=phylum_colors)+
    theme_minimal(base_size = 5) +  
    theme(legend.position = "none") +  # remove legend
    labs(title = paste("Perplexity =", p))
})


ggpubr::ggarrange(plotlist = plots, ncol = 3, nrow = 3) 

```

```{r rows in function table}
nrow(function_table)
```



```{r function_ordination_tsne, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
# Generate the tSNE ordination
set.seed(1001)
tSNE_function <- Rtsne(X=function_table, dims = 2, perplexity = 10, check_duplicates = FALSE)

# Plot the ordination
function_ordination <- tSNE_function$Y %>%
                as.data.frame() %>%
                mutate(genome=rownames(function_table)) %>%
                inner_join(genome_metadata, by="genome") %>%
                rename(tSNE1="V1", tSNE2="V2") %>%
                select(genome,phylum,tSNE1,tSNE2, length) %>%
                ggplot(aes(x = tSNE1, y = tSNE2, color = phylum, size=length))+
                            geom_point(shape=16, alpha=0.7) +
                            scale_color_manual(values=phylum_colors) +
                            theme_minimal() +
                labs(color="Phylum", size="Genome size") +
                guides(color = guide_legend(override.aes = list(size = 5))) # enlarge Phylum dots in legend

function_ordination
```




<!--chapter:end:03_mag_catalogue.Rmd-->

# Community composition

```{r load_data_community, echo=FALSE}
load("data/data.Rdata")
```

## Taxonomy overview 

### Stacked barplot

```{r taxonomy_barplot_location, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0) %>% #filter 0 counts
  ggplot(., aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_grid(. ~ gut_location,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```


```{r taxonomy_barplot, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0) %>% #filter 0 counts
  ggplot(., aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~ gut_location + fox_behaviour,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```

```{r taxonomy_barplot_2, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0) %>% #filter 0 counts
  ggplot(., aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~  fox_behaviour + gut_location,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```

```{r taxonomy_barplot_3, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0) %>% #filter 0 counts
  ggplot(., aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~  fox_behaviour + sample_type,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```

```{r taxonomy_barplot_4, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0) %>% #filter 0 counts
  ggplot(., aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~   sample_type+ fox_behaviour,  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x =element_blank(),
          axis.title.x = element_blank(),
          axis.text.y = element_text(size=8),
          axis.title.y = element_text(size=12),
          panel.background = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black"),
          panel.border = element_rect(colour = "black", fill = NA),
          strip.background = element_rect(fill = "white", color = "black"),
          strip.text = element_text(size = 12, lineheight = 0.6)) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```


### Phylum relative abundances

```{r taxonomy_phylum_summary, warning=FALSE, comments="", message=FALSE}
phylum_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,phylum,fox_behaviour) %>%
  summarise(relabun=sum(count))

phylum_summary %>%
    group_by(phylum) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              tame_mean=mean(relabun[fox_behaviour =="tame"]*100, na.rm=T),
              tame_sd=sd(relabun[fox_behaviour =="tame"]*100, na.rm=T),
              aggressive_mean=mean(relabun[fox_behaviour =="aggr"]*100, na.rm=T),
              aggressive_sd=sd(relabun[fox_behaviour=="aggr"]*100, na.rm=T)) %>%
    mutate(total=str_c(round(total_mean,2),"±",round(total_sd,2)),
           tame=str_c(round(tame_mean,2),"±",round(tame_sd,2)),
           aggresive=str_c(round(aggressive_mean,2),"±",round(aggressive_sd,2))) %>% 
    arrange(-total_mean) %>% 
    select(phylum,total,tame,aggresive) %>% 
    tt()
```

```{r taxonomy_boxplot_phylum, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_arrange <- phylum_summary %>%
    group_by(phylum) %>%
    summarise(mean=mean(relabun)) %>%
    arrange(-mean) %>%
    dplyr::select(phylum) %>%
    pull()

phylum_summary %>%
  filter(phylum %in% phylum_arrange) %>%
  mutate(phylum=factor(phylum,levels=rev(phylum_arrange))) %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
    filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=phylum, group=phylum, color=phylum, fill=phylum)) +
  scale_color_manual(values=phylum_colors[rev(phylum_arrange)]) +
        scale_fill_manual(values=phylum_colors[-8]) +
        geom_boxplot(alpha=0.2)+
        geom_jitter(alpha=0.5) + 
        facet_nested(. ~ gut_location)+ 
        theme_minimal() + 
        theme(legend.position="none",
              strip.text.x = element_text(size = 14, color="black",face="bold"),
              axis.text.x = element_text(vjust = 0.5, size = 10),
              axis.text.y = element_text(size = 12),
              axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
              axis.title = element_text(size = 14, face = "bold"),
              axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
              axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
              ) +
        labs(y="Phylum",x="Relative abundance")
```

```{r taxonomy_phylum_plot, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_summary %>%
  filter(phylum %in% phylum_arrange) %>%
  mutate(phylum=factor(phylum,levels=rev(phylum_arrange))) %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
    filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=phylum, group=phylum, color=phylum, fill=phylum)) +
  scale_color_manual(values=phylum_colors[rev(phylum_arrange)]) +
        scale_fill_manual(values=phylum_colors[-8]) +
        geom_boxplot(alpha=0.2)+
        geom_jitter(alpha=0.5) + 
        theme_minimal() + 
        theme(legend.position="none") +
        labs(y="Phylum",x="Relative abundance")
```

### Family

```{r family_all, comment="", echo=FALSE}
family_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalization
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,family,fox_behaviour) %>%
  summarise(relabun=sum(count))

family_arrange <- family_summary %>%
    group_by(family) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    dplyr::select(family) %>%
    pull()

family_summary %>%
    group_by(family) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              tame_mean=mean(relabun[fox_behaviour=="tame"]*100, na.rm=T),
              tame_sd=sd(relabun[fox_behaviour=="tame"]*100, na.rm=T),
              aggr_mean=mean(relabun[fox_behaviour=="aggr"]*100, na.rm=T),
              aggr_sd=sd(relabun[fox_behaviour=="aggr"]*100, na.rm=T)) %>%
    mutate(total=str_c(round(total_mean,2),"±",round(total_sd,2)),
           tame=str_c(round(tame_mean,2),"±",round(tame_sd,2)),
           aggr=str_c(round(aggr_mean,2),"±",round(aggr_sd,2))) %>% 
    arrange(-total_mean) %>% 
    select(family,total,tame,aggr) %>% 
    tt()
```



```{r taxonomy_jitterplot_family_location, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
family_summary %>%
  left_join(genome_metadata %>% dplyr::select(family,phylum) %>% unique(),by=join_by(family==family)) %>%
  left_join(sample_metadata,by=join_by(sample==sample)) %>%
  filter(family %in% family_arrange[1:20]) %>%
  mutate(family=factor(family,levels=rev(family_arrange[1:20]))) %>%
  filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=family, group=family, color=phylum, fill=phylum)) +
      scale_color_manual(values=phylum_colors[-8]) +
      scale_fill_manual(values=phylum_colors[-8]) +
      geom_boxplot(alpha=0.2)+
      geom_jitter(alpha=0.5) + 
      facet_grid(.~gut_location)+
      theme_minimal() +
      theme(legend.position="none",
                  strip.text.x = element_text(size = 14, color="black",face="bold"),
                  axis.text.x = element_text(vjust = 0.5, size = 6),
                  axis.text.y = element_text(size = 12),
                  axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
                  axis.title = element_text(size = 14, face = "bold"),
                  axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
                  axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)))+
      labs(y="Family", x="Relative abundance", color="Phylum")
```

```{r taxonomy_jitterplot_family_origin, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
family_summary %>%
    left_join(genome_metadata %>% dplyr::select(family,phylum) %>% unique(),by=join_by(family==family)) %>%
    left_join(sample_metadata,by=join_by(sample==sample)) %>%
    filter(family %in% family_arrange[1:20]) %>%
    mutate(family=factor(family,levels=rev(family_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=family, group=family, color=phylum, fill=phylum)) +
        scale_color_manual(values=phylum_colors[-8]) +
        scale_fill_manual(values=phylum_colors[-8]) +
        geom_boxplot(alpha=0.2)+
        geom_jitter(alpha=0.5) + 
        facet_grid(.~gut_location)+
        theme_minimal() +
        theme(legend.position="none",
                    strip.text.x = element_text(size = 14, color="black",face="bold"),
                    axis.text.x = element_text(vjust = 0.5, size = 6),
                    axis.text.y = element_text(size = 12),
                    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
                    axis.title = element_text(size = 14, face = "bold"),
                    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
                    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)))+
        labs(y="Family", x="Relative abundance", color="Phylum")
```

### Genus

```{r genus_all, comment="", echo=FALSE}
genus_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,genus,fox_behaviour) %>%
  summarise(relabun=sum(count))

genus_arrange <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=sum(relabun)) %>%
    filter(genus != "g__")%>%
    arrange(-mean) %>%
    dplyr::select(genus) %>%
    mutate(genus= sub("^g__", "", genus)) %>%
    pull()

genus_summary %>%
    group_by(genus) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              tame_mean=mean(relabun[fox_behaviour=="tame"]*100, na.rm=T),
              tame_sd=sd(relabun[fox_behaviour=="tame"]*100, na.rm=T),
              aggr_mean=mean(relabun[fox_behaviour=="aggr"]*100, na.rm=T),
              aggr_sd=sd(relabun[fox_behaviour=="aggr"]*100, na.rm=T)) %>%
    mutate(total=str_c(round(total_mean,2),"±",round(total_sd,2)),
           tame=str_c(round(tame_mean,2),"±",round(tame_sd,2)),
           aggr=str_c(round(aggr_mean,2),"±",round(aggr_sd,2))) %>% 
    arrange(-total_mean) %>% 
    select(genus,total,tame,aggr) %>% 
    tt()
```

```{r taxonomy_jitterplot_genus_location, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genus_summary %>%
  left_join(genome_metadata %>% dplyr::select(genus,phylum) %>% unique(),by="genus") %>%
  left_join(sample_metadata,by=join_by(sample==sample)) %>%
  filter(genus %in% genus_arrange[1:20]) %>%
  mutate(genus=factor(genus,levels=rev(genus_arrange[1:20]))) %>%
  filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=genus, group=genus, color=phylum, fill=phylum)) +
  scale_color_manual(values=phylum_colors[-8]) +
  scale_fill_manual(values=phylum_colors[-8]) +
  geom_boxplot(alpha=0.2)+
  geom_jitter(alpha=0.5) + 
  facet_grid(.~ gut_location)+
  theme_minimal() +
  theme(legend.position="none",
              strip.text.x = element_text(size = 14, color="black",face="bold"),
              axis.text.x = element_text(vjust = 0.5, size = 6),
              axis.text.y = element_text(size = 12),
              axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
              axis.title = element_text(size = 14, face = "bold"),
              axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
              axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)))+
  labs(y="Family", x="Relative abundance", color="Phylum")
```

```{r taxonomy_jitterplot_genus_origin, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
genus_summary %>%
  left_join(genome_metadata %>% dplyr::select(genus,phylum) %>% unique(),by="genus") %>%
  filter(genus %in% genus_arrange[1:20]) %>%
  mutate(genus=factor(genus,levels=rev(genus_arrange[1:20]))) %>%
  filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=genus, group=genus, color=phylum, fill=phylum)) +
  scale_color_manual(values=phylum_colors[-8]) +
  scale_fill_manual(values=phylum_colors[-8]) +
  geom_boxplot(alpha=0.2)+
  geom_jitter(alpha=0.5) + 
  facet_grid(.~fox_behaviour)+
  theme_minimal() +
  theme(legend.position="none",
              strip.text.x = element_text(size = 14, color="black",face="bold"),
              axis.text.x = element_text(vjust = 0.5, size = 6),
              axis.text.y = element_text(size = 12),
              axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
              axis.title = element_text(size = 14, face = "bold"),
              axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
              axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0)))+
  labs(y="Family", x="Relative abundance", color="Phylum")
```

## Core microbiota

```{r core_mag_upset, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
library(UpSetR)

genome_counts_rel <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>%
  column_to_rownames(., "genome")

#Presence/absence
genome_counts_rel_pa <- (genome_counts_rel > 0) * 1

#Add gut_location info
df <- as.data.frame(t(genome_counts_rel_pa)) %>%
  rownames_to_column("sample") %>%
  left_join(sample_metadata, by="sample") 

#Summarize by location
table_upset_analysis <- df %>%
  group_by(gut_location) %>%
  summarise(across(-sample, ~ as.integer(any(. > 0)))) %>%
  column_to_rownames("gut_location") %>%
  t() %>%
  as.data.frame()

#UpSet plot
upset(table_upset_analysis,
      keep.order = TRUE,
      sets = rev(c("F_colon", "M_colon", "D_ileum", "K_ileum")),
      sets.bar.color = rev(location_colors[1:4]),
      mb.ratio = c(0.55, 0.45),
      order.by = "freq")

```

```{r core_mag_origin, warning=FALSE, comments="", message=FALSE}
mag_prevalence_location <- genome_counts_filt  %>%
  mutate_at(vars(-genome),~./sum(.)) %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  mutate(presence=ifelse(abundance>0,1,0)) %>% 
  select(genome,sample,presence,gut_location, sample_type) %>%
  group_by(genome,gut_location, sample_type) %>%
  summarise(presence=ifelse(sum(presence)>0,1,0)) %>% 
  group_by(genome,sample_type) %>%
  summarise(presence=sum(presence))

mag_prevalence_location %>% 
  group_by(sample_type) %>%
  summarise(mean=mean(presence),sd=sd(presence)) %>%
  tt()

wilcox.test(presence ~ sample_type, data=mag_prevalence_location)  %>%
  tidy()
```

```{r core_mag_origin_boxplot, warning=FALSE, comments="", message=FALSE}
mag_prevalence_location %>% 
  ggplot(aes(x=sample_type,y=presence, color=sample_type, fill=sample_type)) +
    geom_boxplot() +
    geom_jitter() +
    scale_color_manual(values=behaviour_colors) +
    scale_fill_manual(values=str_c(behaviour_colors,"50")) +
    theme_minimal()
```

```{r core_mag_fox_behaviour, warning=FALSE, comments="", message=FALSE}
mag_prevalence_location <- genome_counts_filt  %>%
  mutate_at(vars(-genome),~./sum(.)) %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  filter(!grepl("unsel", fox_behaviour))%>% #to remove the rows with "unsel"
  mutate(presence=ifelse(abundance>0,1,0)) %>% 
  select(genome,sample,presence,gut_location, fox_behaviour) %>%
  group_by(genome,gut_location, fox_behaviour) %>%
  summarise(presence=ifelse(sum(presence)>0,1,0)) %>% 
  group_by(genome,fox_behaviour) %>%
  summarise(presence=sum(presence))

mag_prevalence_location %>% 
  group_by(fox_behaviour) %>%
  summarise(mean=mean(presence),sd=sd(presence)) %>%
  tt()

wilcox.test(presence ~ fox_behaviour, data=mag_prevalence_location)  %>%
  tidy()
```

```{r core_mag_behaviour_boxplot, warning=FALSE, comments="", message=FALSE}
behaviour_colors <- c( "#d62728", "#1f77b4","#2ca02c")

mag_prevalence_location %>% 
  ggplot(aes(x=fox_behaviour,y=presence, color=fox_behaviour, fill=fox_behaviour)) +
    geom_boxplot() +
    geom_jitter() +
    scale_color_manual(values=behaviour_colors) +
    scale_fill_manual(values=str_c(behaviour_colors,"50")) +
    theme_minimal()
```

<!--chapter:end:04_community_composition.Rmd-->

# Alpha diversity

```{r load_data_alpha, comment="", message=FALSE, warning=FALSE}
load("data/data.Rdata")
```


## Summary table

```{r alpha_div, comment="", message=FALSE, warning=FALSE}
behaviour_colors <- c(
  tame = "#1f77b4",   
  aggr = "#d62728",   
  unsel = "#2ca02c"   
)

# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")

# # Aggregate basal GIFT into elements
genome_counts_filt <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt$genome,]
genome_gifts <- genome_gifts[, colSums(genome_gifts != 0) > 0]

dist <- genome_gifts %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

functional <- genome_counts_filt %>%
  filter(genome %in% rownames(dist)) %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, dist = dist) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(functional = 1) %>%
  rownames_to_column(var = "sample") %>%
  mutate(functional = if_else(is.nan(functional), 1, functional))

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample))%>%
  full_join(functional, by = join_by(sample == sample))
```

```{r alpha_div_fox_behaviours_summary_all, comment="",echo=FALSE, message=FALSE, warning=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
    group_by(alpha)%>%
    summarise(
              aggr_mean=mean(value[fox_behaviour=="aggr"], na.rm=T),
              aggr_sd=sd(value[fox_behaviour=="aggr"], na.rm=T),
              tame_mean=mean(value[fox_behaviour=="tame"], na.rm=T),
              tame_sd=sd(value[fox_behaviour=="tame"], na.rm=T),
              unsel_mean=mean(value[fox_behaviour=="unsel"], na.rm=T),
              unsel_sd=sd(value[fox_behaviour=="unsel"], na.rm=T)) %>%
    mutate(
           aggr=str_c(round(aggr_mean,2),"±",round(aggr_sd,2)),
           tame=str_c(round(tame_mean,2),"±",round(tame_sd,2)),
           unsel=str_c(round(unsel_mean,2),"±",round(unsel_sd,2))) %>% 
    arrange(-aggr_mean) %>% 
    dplyr::select(alpha,aggr,tame,unsel) %>% 
    tt()
```

## Aggressive vs Tame

### Shapiro test

```{r alpha_div_shapiro, comment="", message=FALSE, warning=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="richness") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="neutral") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="phylogenetic") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="phylogenetic") %>% 
  summarize(var.test_p_value_phylo = var.test(value ~ fox_behaviour)$p.value) 
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="functional") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="functional") %>% 
  summarize(var.test_p_functional = var.test(value ~ fox_behaviour)$p.value) 
```

### Plots

```{r alpha_div_boxplot, comment="",echo=FALSE, message=FALSE, warning=FALSE}
sample_metadata$fox_behaviour <- factor(sample_metadata$fox_behaviour, levels=c("tame", "unsel", "aggr"))
#Richness
plot1 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="richness") %>%
   filter(!fox_behaviour =="unsel") %>% 
      ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color= fox_behaviour, fill=fox_behaviour)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
  scale_x_discrete(labels=c("aggr" = "Aggressive", "tame" = "Tame", "unsel" = "Unselected")) +
    stat_compare_means(method = "wilcox.test", show.legend = F, size = 3, label.y = c(850), label.x = c(1.5))+
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Richness")

#Neutral
plot2 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="neutral") %>%
   filter(!fox_behaviour =="unsel") %>% 
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
  scale_x_discrete(labels=c("aggr" = "Aggressive", "tame" = "Tame",  "unsel" = "Unselected")) +
  stat_compare_means(method = "wilcox.test",show.legend = F, size = 3, label.y = c(150), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  coord_cartesian(ylim = c(0, 150)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Neutral")

#Phylogenetic
plot3 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="phylogenetic") %>%
   filter(!fox_behaviour =="unsel") %>% 
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
  scale_x_discrete(labels=c("aggr" = "Aggressive", "tame" = "Tame",  "unsel" = "Unselected")) +
  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(11), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Phylogenetic")

#Functional
plot4 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="functional") %>%
   filter(!fox_behaviour =="unsel") %>% 
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
  scale_x_discrete(labels=c("aggr" = "Aggresive", "tame" = "Tame",  "unsel" = "Unselected")) +
  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Functional")

```

```{r div_plot_together, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot1,plot2,plot3, plot4, ncol = 2))
```
```{r div_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=8, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot2,plot3))
```





## Tame vs Unselected
```{r alpha_div_fox_behaviours_summary, comment="",echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
    group_by(alpha)%>%
    summarise(
              tame_mean=mean(value[fox_behaviour=="tame"], na.rm=T),
              tame_sd=sd(value[fox_behaviour=="tame"], na.rm=T),
              unsel_mean=mean(value[fox_behaviour=="unsel"], na.rm=T),
              unsel_sd=sd(value[fox_behaviour=="unsel"], na.rm=T)) %>%
    mutate(
           tame=str_c(round(tame_mean,2),"±",round(tame_sd,2)),
           unsel=str_c(round(unsel_mean,2),"±",round(unsel_sd,2))) %>% 
    arrange(-tame_mean) %>% 
    dplyr::select(alpha,tame,unsel) %>% 
    tt()
```

### Plots
```{r alpha_div_tame_unsel_boxplot, comment="",echo=FALSE, message=FALSE, warning=FALSE}
sample_metadata$fox_behaviour <- factor(sample_metadata$fox_behaviour, levels=c("tame", "unsel", "aggr"))
#Richness
plot1 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="aggr") %>%
  filter(metric=="richness") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "wilcox.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Richness")

#Neutral
plot2 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="aggr") %>%
  filter(metric=="neutral") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  coord_cartesian(ylim = c(0, 150)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Neutral")

#Phylogenetic
plot3 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="aggr") %>% 
  filter(metric=="phylogenetic") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Phylogenetic")

#Functional
plot4 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="aggr") %>%
  filter(metric=="functional") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "wilcox.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Functional")
```
```{r div_plot_tame_unsel_all, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot1,plot2,plot3, plot4, ncol = 2))
```
```{r div_tame_unsel_post2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=6}
grid.arrange(arrangeGrob(plot2,plot3))
```


## Aggressive vs Unselected:

```{r alpha_div_fox_behaviours_aggr_unsel, comment="",echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
    group_by(alpha)%>%
    summarise(
              aggr_mean=mean(value[fox_behaviour=="aggr"], na.rm=T),
              aggr_sd=sd(value[fox_behaviour=="aggr"], na.rm=T),
              unsel_mean=mean(value[fox_behaviour=="unsel"], na.rm=T),
              unsel_sd=sd(value[fox_behaviour=="unsel"], na.rm=T)) %>%
    mutate(
           aggr=str_c(round(aggr_mean,2),"±",round(aggr_sd,2)),
           unsel=str_c(round(unsel_mean,2),"±",round(unsel_sd,2))) %>% 
    arrange(-aggr_mean) %>% 
    dplyr::select(alpha,aggr,unsel) %>% 
    tt()
```

### Plots
```{r alpha_div_aggrvsunsel_boxplot, comment="",echo=FALSE, message=FALSE, warning=FALSE}
sample_metadata$fox_behaviour <- factor(sample_metadata$fox_behaviour, levels=c("tame", "unsel", "aggr"))
#Richness
plot1 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="tame") %>%
  filter(metric=="richness") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "wilcox.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Richness")

#Neutral
plot2 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="tame") %>%
  filter(metric=="neutral") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  coord_cartesian(ylim = c(0, 150)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Neutral")

#Phylogenetic
plot3 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="tame") %>% 
  filter(metric=="phylogenetic") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Phylogenetic")

#Functional
plot4 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="tame") %>%
  filter(metric=="functional") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "wilcox.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Functional")
```
```{r div_plot_all_unsel_aggr, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot1,plot2,plot3, plot4, ncol = 2))
```

```{r div_plot_pre_post2_unsel_aggr, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=6}
grid.arrange(arrangeGrob(plot2,plot3))
```

```{r Testing 3 at once}
alpha_long <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample))

# Shapiro–Wilk test (normality per group and metric)
shapiro_results <- alpha_long %>%
  group_by(metric, fox_behaviour) %>%
  shapiro_test(value)

# Levene’s test (variance homogeneity across groups, per metric)
levene_results <- alpha_long %>%
  group_by(metric) %>%
  levene_test(value ~ fox_behaviour)

# Kruskal–Wallis test (non-parametric ANOVA alternative, 3+ groups)
kruskal_results <- alpha_long %>%
  group_by(metric) %>%
  kruskal_test(value ~ fox_behaviour)

# Pairwise comparisons (Dunn test with p.adjust)
pairwise_results <- alpha_long %>%
  group_by(metric) %>%
  dunn_test(value ~ fox_behaviour, p.adjust.method = "bonferroni")

# View results
shapiro_results
levene_results
kruskal_results
pairwise_results

```
```{r Plotting all together}
# Ensure factor order
alpha_long <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  mutate(fox_behaviour = factor(fox_behaviour, levels = c("tame", "unsel", "aggr")))

# Run Kruskal-Wallis tests per metric
kw_results <- alpha_long %>%
  group_by(metric) %>%
  kruskal_test(value ~ fox_behaviour)

# Build custom facet labels with p-values
facet_labels <- kw_results %>%
  mutate(label = paste0(metric, " (KW p=", signif(p, 3), ")")) %>%
  select(metric, label) %>%
  deframe()

# Plot
plot_all <- alpha_long %>%
  ggplot(aes(x = fox_behaviour, y = value,
             color = fox_behaviour, fill = fox_behaviour)) +
  geom_jitter(width = 0.2, alpha = 0.7, size = 2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha = 0.3, outlier.shape = NA, show.legend = FALSE) +
  facet_wrap(~ metric, scales = "free_y", ncol = 2,
             labeller = labeller(metric = facet_labels)) +
  scale_color_manual(values = behaviour_colors) +
  scale_fill_manual(values = behaviour_colors) +
  theme_classic(base_size = 12) +
  theme(
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    axis.title.y = element_text(margin = margin(r = 15)),
    axis.title.x = element_text(margin = margin(t = 15))
  ) +
  labs(x = "Origin", y = "Alpha diversity")

plot_all

```




<!--chapter:end:05_alpha_diversity.Rmd-->

# Alpha diversity

```{r load_data_alpha_2, comment="", message=FALSE, warning=FALSE}
load("data/data.Rdata")
```

```{r filter samples_2}
sample_metadata <- sample_metadata %>%
  filter(gut_location == "F_colon")

common_samples <- intersect(sample_metadata$sample, colnames(genome_counts_filt))

sample_metadata <- sample_metadata %>%
  filter(sample %in% common_samples)

genome_counts_filt <- genome_counts_filt %>%
  select(all_of(common_samples), genome)
  
```


## Summary table

```{r alpha_div_2, comment="", message=FALSE, warning=FALSE}
behaviour_colors <- c(
  tame = "#1f77b4",   
  aggr = "#d62728",   
  unsel = "#2ca02c"   
)

# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")

# # Aggregate basal GIFT into elements
genome_counts_filt <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt$genome,]
genome_gifts <- genome_gifts[, colSums(genome_gifts != 0) > 0]

dist <- genome_gifts %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

functional <- genome_counts_filt %>%
  filter(genome %in% rownames(dist)) %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, dist = dist) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(functional = 1) %>%
  rownames_to_column(var = "sample") %>%
  mutate(functional = if_else(is.nan(functional), 1, functional))

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample))%>%
  full_join(functional, by = join_by(sample == sample))
```

```{r alpha_div_fox_behaviours_summary_all_2, comment="",echo=FALSE, message=FALSE, warning=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
    group_by(alpha)%>%
    summarise(
              aggr_mean=mean(value[fox_behaviour=="aggr"], na.rm=T),
              aggr_sd=sd(value[fox_behaviour=="aggr"], na.rm=T),
              tame_mean=mean(value[fox_behaviour=="tame"], na.rm=T),
              tame_sd=sd(value[fox_behaviour=="tame"], na.rm=T),
              unsel_mean=mean(value[fox_behaviour=="unsel"], na.rm=T),
              unsel_sd=sd(value[fox_behaviour=="unsel"], na.rm=T)) %>%
    mutate(
           aggr=str_c(round(aggr_mean,2),"±",round(aggr_sd,2)),
           tame=str_c(round(tame_mean,2),"±",round(tame_sd,2)),
           unsel=str_c(round(unsel_mean,2),"±",round(unsel_sd,2))) %>% 
    arrange(-aggr_mean) %>% 
    dplyr::select(alpha,aggr,tame,unsel) %>% 
    tt()
```

## Aggressive vs Tame

### Shapiro test

```{r alpha_div_shapiro_2, comment="", message=FALSE, warning=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="richness") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="neutral") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="phylogenetic") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="phylogenetic") %>% 
  summarize(var.test_p_value_phylo = var.test(value ~ fox_behaviour)$p.value) 
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="functional") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="unsel") %>% 
  filter(metric=="functional") %>% 
  summarize(var.test_p_functional = var.test(value ~ fox_behaviour)$p.value) 
```

### Plots

```{r alpha_div_boxplot_2, comment="",echo=FALSE, message=FALSE, warning=FALSE}
sample_metadata$fox_behaviour <- factor(sample_metadata$fox_behaviour, levels=c("tame", "unsel", "aggr"))
#Richness
plot1 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="richness") %>%
   filter(!fox_behaviour =="unsel") %>% 
      ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color= fox_behaviour, fill=fox_behaviour)) +
      geom_jitter(width = 0.2, show.legend = FALSE) +
      geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
  scale_x_discrete(labels=c("aggr" = "Aggressive", "tame" = "Tame", "unsel" = "Unselected")) +
    stat_compare_means(method = "wilcox.test", show.legend = F, size = 3, label.y = c(850), label.x = c(1.5))+
coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Richness")

#Neutral
plot2 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="neutral") %>%
   filter(!fox_behaviour =="unsel") %>% 
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
  scale_x_discrete(labels=c("aggr" = "Aggressive", "tame" = "Tame",  "unsel" = "Unselected")) +
  stat_compare_means(method = "wilcox.test",show.legend = F, size = 3, label.y = c(150), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  coord_cartesian(ylim = c(0, 150)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Neutral")

#Phylogenetic
plot3 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="phylogenetic") %>%
   filter(!fox_behaviour =="unsel") %>% 
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
  scale_x_discrete(labels=c("aggr" = "Aggressive", "tame" = "Tame",  "unsel" = "Unselected")) +
  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(11), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Phylogenetic")

#Functional
plot4 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="functional") %>%
   filter(!fox_behaviour =="unsel") %>% 
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
  scale_x_discrete(labels=c("aggr" = "Aggresive", "tame" = "Tame",  "unsel" = "Unselected")) +
  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Origin", y = "Functional")

```

```{r div_plot_together_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot1,plot2,plot3, plot4, ncol = 2))
```
```{r div_plot_2_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=8, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot2,plot3))
```





## Tame vs Unselected
```{r alpha_div_fox_behaviours_summary_2, comment="",echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
    group_by(alpha)%>%
    summarise(
              tame_mean=mean(value[fox_behaviour=="tame"], na.rm=T),
              tame_sd=sd(value[fox_behaviour=="tame"], na.rm=T),
              unsel_mean=mean(value[fox_behaviour=="unsel"], na.rm=T),
              unsel_sd=sd(value[fox_behaviour=="unsel"], na.rm=T)) %>%
    mutate(
           tame=str_c(round(tame_mean,2),"±",round(tame_sd,2)),
           unsel=str_c(round(unsel_mean,2),"±",round(unsel_sd,2))) %>% 
    arrange(-tame_mean) %>% 
    dplyr::select(alpha,tame,unsel) %>% 
    tt()
```

### Plots
```{r alpha_div_tame_unsel_boxplot_2, comment="",echo=FALSE, message=FALSE, warning=FALSE}
sample_metadata$fox_behaviour <- factor(sample_metadata$fox_behaviour, levels=c("tame", "unsel", "aggr"))
#Richness
plot1 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="aggr") %>%
  filter(metric=="richness") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "wilcox.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Richness")

#Neutral
plot2 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="aggr") %>%
  filter(metric=="neutral") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  coord_cartesian(ylim = c(0, 150)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Neutral")

#Phylogenetic
plot3 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="aggr") %>% 
  filter(metric=="phylogenetic") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Phylogenetic")

#Functional
plot4 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="aggr") %>%
  filter(metric=="functional") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "wilcox.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Functional")
```
```{r div_plot_tame_unsel_all_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot1,plot2,plot3, plot4, ncol = 2))
```
```{r div_tame_unsel_post2_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=6}
grid.arrange(arrangeGrob(plot2,plot3))
```


## Aggressive vs Unselected

```{r alpha_div_fox_behaviours_aggr_unsel_2, comment="",echo=FALSE, message=FALSE, warning=FALSE, eval=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
    group_by(alpha)%>%
    summarise(
              aggr_mean=mean(value[fox_behaviour=="aggr"], na.rm=T),
              aggr_sd=sd(value[fox_behaviour=="aggr"], na.rm=T),
              unsel_mean=mean(value[fox_behaviour=="unsel"], na.rm=T),
              unsel_sd=sd(value[fox_behaviour=="unsel"], na.rm=T)) %>%
    mutate(
           aggr=str_c(round(aggr_mean,2),"±",round(aggr_sd,2)),
           unsel=str_c(round(unsel_mean,2),"±",round(unsel_sd,2))) %>% 
    arrange(-aggr_mean) %>% 
    dplyr::select(alpha,aggr,unsel) %>% 
    tt()
```

### Plots
```{r alpha_div_aggrvsunsel_boxplot_2, comment="",echo=FALSE, message=FALSE, warning=FALSE}
sample_metadata$fox_behaviour <- factor(sample_metadata$fox_behaviour, levels=c("tame", "unsel", "aggr"))
#Richness
plot1 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="tame") %>%
  filter(metric=="richness") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "wilcox.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Richness")

#Neutral
plot2 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="tame") %>%
  filter(metric=="neutral") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  coord_cartesian(ylim = c(0, 150)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Neutral")

#Phylogenetic
plot3 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="tame") %>% 
  filter(metric=="phylogenetic") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Phylogenetic")

#Functional
plot4 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(!fox_behaviour =="tame") %>%
  filter(metric=="functional") %>%
  ggplot(aes(y = value, x = fox_behaviour, group=fox_behaviour, color=fox_behaviour, fill=fox_behaviour)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  scale_color_manual(values=behaviour_colors)+
  scale_fill_manual(values=behaviour_colors) +
#  stat_compare_means(method = "wilcox.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "fox_behaviour", y = "Functional")
```
```{r div_plot_all_unsel_aggr_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot1,plot2,plot3, plot4, ncol = 2))
```

```{r div_plot_2_unsel_aggr_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=6}
grid.arrange(arrangeGrob(plot2,plot3))
```

```{r Testing 3 at once_2}
alpha_long <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample))

# Shapiro–Wilk test (normality per group and metric)
shapiro_results <- alpha_long %>%
  group_by(metric, fox_behaviour) %>%
  shapiro_test(value)

# Levene’s test (variance homogeneity across groups, per metric)
levene_results <- alpha_long %>%
  group_by(metric) %>%
  levene_test(value ~ fox_behaviour)

# Kruskal–Wallis test (non-parametric ANOVA alternative, 3+ groups)
kruskal_results <- alpha_long %>%
  group_by(metric) %>%
  kruskal_test(value ~ fox_behaviour)

# Pairwise comparisons (Dunn test with p.adjust)
pairwise_results <- alpha_long %>%
  group_by(metric) %>%
  dunn_test(value ~ fox_behaviour, p.adjust.method = "bonferroni")

# View results
shapiro_results
levene_results
kruskal_results
pairwise_results

```
```{r Plotting all together_2}
# Ensure factor order
alpha_long <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  mutate(fox_behaviour = factor(fox_behaviour, levels = c("tame", "unsel", "aggr")))

# Run Kruskal-Wallis tests per metric
kw_results <- alpha_long %>%
  group_by(metric) %>%
  kruskal_test(value ~ fox_behaviour)

# Build custom facet labels with p-values
facet_labels <- kw_results %>%
  mutate(label = paste0(metric, " (KW p=", signif(p, 3), ")")) %>%
  select(metric, label) %>%
  deframe()

# Plot
plot_all <- alpha_long %>%
  ggplot(aes(x = fox_behaviour, y = value,
             color = fox_behaviour, fill = fox_behaviour)) +
  geom_jitter(width = 0.2, alpha = 0.7, size = 2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha = 0.3, outlier.shape = NA, show.legend = FALSE) +
  facet_wrap(~ metric, scales = "free_y", ncol = 2,
             labeller = labeller(metric = facet_labels)) +
  scale_color_manual(values = behaviour_colors) +
  scale_fill_manual(values = behaviour_colors) +
  theme_classic(base_size = 12) +
  theme(
    strip.background = element_blank(),
    strip.text = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    axis.title.y = element_text(margin = margin(r = 15)),
    axis.title.x = element_text(margin = margin(t = 15))
  ) +
  labs(x = "Origin", y = "Alpha diversity")

plot_all

```




<!--chapter:end:05_alpha_diversity_only_F_colon.Rmd-->

# Beta diversity

```{r load_data_beta_diversity}
load("data/data.Rdata")

behaviour_colors <- c(
  tame = "#1f77b4",   
  aggr = "#d62728",   
  unsel = "#2ca02c"   
)
```

```{r prepare sample_metadata}
sample_metadata <- sample_metadata %>%
  filter(sample %in% colnames(genome_counts_filt))
```


```{r beta_div, comment="", message=FALSE, warning=FALSE, eval=FALSE}
beta_q0n <- genome_counts_filt %>% 
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1)

genome_counts_filt_beta <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_tree <- keep.tip(genome_tree, tip=genome_counts_filt_beta$genome)
beta_q1p <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1, tree = genome_tree)

genome_gifts1 <- genome_gifts[genome_counts_filt_beta$genome,]
genome_gifts1 <- genome_gifts1[, colSums(genome_gifts1 != 0) > 0]

dist <- genome_gifts1 %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

beta_q1f <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1, dist = dist)
```

```{r save_beta, comment="", message=FALSE,echo=FALSE,warning=FALSE, eval=FALSE}
save(beta_q0n, 
     beta_q1n, 
     beta_q1p, 
     beta_q1f, 
     file = "data/beta.Rdata")
```

```{r load beta data}
set.seed(2024)
load("data/beta.Rdata")
```

### Permanova

```{r permanova, comment="", message=FALSE, warning=FALSE}
#Richness

betadisper(beta_q0n$S, sample_metadata$fox_behaviour) %>% permutest(.) 

adonis2(beta_q0n$S ~ fox_behaviour+gut_location,
        by="terms", 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q0n$S))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r permanova_neutral, comment="", message=FALSE, warning=FALSE}
#Neutral diversity
betadisper(beta_q1n$S, sample_metadata$fox_behaviour) %>% permutest(.) 
adonis2(beta_q1n$S ~ fox_behaviour +gut_location,
        by="terms",
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1n$S))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r pairwise_neutral, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(beta_q1n$S, sample_metadata$fox_behaviour, perm = 999)
```

```{r permanova_phylo, comment="", message=FALSE, warning=FALSE}
#Phylogenetic diversity
betadisper(beta_q1p$S, sample_metadata$fox_behaviour) %>% permutest(.) 
adonis2(beta_q1p$S ~ fox_behaviour+ gut_location,
        by="terms",
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1p$S))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r permanova_func, comment="", message=FALSE, warning=FALSE}
#Functional diversity
betadisper(beta_q1f$S, sample_metadata$fox_behaviour) %>% permutest(.) 
adonis2(beta_q1f$S ~ fox_behaviour+ gut_location,
        by="terms", 
        data = sample_metadata %>% arrange(match(sample,labels(beta_q1f$S))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

```{r pairwise_func, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(beta_q1f$S, sample_metadata$sample_type, perm = 999)
```
```{r pairwise_func_location, comment="", message=FALSE, warning=FALSE}
pairwise.adonis(beta_q1f$S, sample_metadata$gut_location, perm = 999)
```

### NMDS
####Richness
```{r beta_div_nmds_richness_plot, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q0n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(fox_behaviour) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = fox_behaviour, shape = as.factor(sample_type))) +
  geom_point(size = 4) +
  scale_color_manual(values = behaviour_colors,labels=c("tame" = "Tame", "aggr" = "Aggressive", "unsel" = "Unselected")) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    ) +
    labs(color="Fox behaviour", shape="Sample Type")+geom_text_repel(aes(label = sample), size=3)

```

#### Richness between gut locations
```{r beta_div_nmds_richness_plot_location, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q0n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(gut_location) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = gut_location, shape = as.factor(fox_behaviour))) +
  geom_point(size = 4) +
  scale_color_manual(values = location_colors) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    ) +
    labs(color="Gut location", shape="Fox Behaviour")+geom_text_repel(aes(label = sample), size=3)

```

#### Neutral diversity

```{r beta_div_nmds_neutral_plot, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q1n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(fox_behaviour) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = fox_behaviour, shape = as.factor(sample_type))) +
  geom_point(size = 4) +
  scale_color_manual(values = behaviour_colors,labels=c("tame" = "Tame", "aggr" = "Aggressive", "unsel" = "Unselected")) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    ) +
    labs(color="Fox Behaviour", shape="Sample Type")
```

####Neutral diversity by gut location
```{r beta_div_nmds_neutral_plot_location, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q1n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(gut_location) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = gut_location, shape = as.factor(fox_behaviour))) +
  geom_point(size = 4) +
  scale_color_manual(values = location_colors) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    ) +
    labs(color="Gut location", shape="Fox Behaviour")
```

#### Phylogenetic diversity

```{r beta_div_nmds_phylo_plot, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q1p$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(fox_behaviour) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = fox_behaviour, shape = as.factor(sample_type))) +
  geom_point(size = 4) +
  scale_color_manual(values = behaviour_colors,labels=c("tame" = "Tame", "aggr" = "Aggressive", "unsel" = "Unselected")) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    ) +
    labs(color="Fox Behaviour", shape="Sample Type")+geom_text_repel(aes(label = sample), size=3)
```

#### Phylogenetic diversity by gut location
```{r beta_div_nmds_phylo_plot_location, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q1p$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(gut_location) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = gut_location, shape = as.factor(fox_behaviour))) +
  geom_point(size = 4) +
  scale_color_manual(values = location_colors) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    ) +
    labs(color="Gut location", shape="Fox Behaviour")+geom_text_repel(aes(label = sample), size=3)
```

#### Functional diversity

```{r beta_div_nmds_funct_plot, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q1f$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(fox_behaviour) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = fox_behaviour, shape = as.factor(sample_type))) +
  geom_point(size = 4) +
  scale_color_manual(values = behaviour_colors,labels=c("tame" = "Tame", "aggr" = "Aggressive", "unsel" = "Unselected")) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    ) +
    labs(color="Fox Behaviour", shape="Sample Type")+geom_text_repel(aes(label = sample), size=3)
```

#### Functional diversity by gut location
```{r beta_div_nmds_funct_plot_location, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE}
beta_q1f$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(gut_location) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = gut_location, shape = as.factor(fox_behaviour))) +
  geom_point(size = 4) +
  scale_color_manual(values = location_colors) +
  geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
  theme(
    axis.title = element_text(size = 12, face = "bold"),
    axis.text = element_text(size = 10),
    panel.background = element_blank(),
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    legend.text = element_text(size = 10),
    legend.title = element_text(size = 12),
    legend.position = "right", legend.box = "vertical"
    ) +
    labs(color="Gut location", shape="Fox Behaviour")+geom_text_repel(aes(label = sample), size=3)
```

<!--chapter:end:06_beta_diversity.Rmd-->

